<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Adding custom tracks • hidecan</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../apple-touch-icon-60x60.png">
<script src="../../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../../deps/headroom-0.11.0/headroom.min.js"></script><script src="../../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../../deps/search-1.0.0/fuse.min.js"></script><script src="../../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../../pkgdown.js"></script><meta property="og:title" content="Adding custom tracks">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../../index.html">hidecan</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../../articles/hidecan.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../../articles/hidecan-step-by-step.html">Step-by-step</a></li>
    <li><a class="dropdown-item" href="../../articles/web_only/gwaspoly_output.html">GWASpoly output</a></li>
    <li><a class="dropdown-item" href="../../articles/web_only/custom_tracks.html">Adding custom tracks</a></li>
    <li><a class="dropdown-item" href="../../articles/web_only/shiny-app.html">Shiny app</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/PlantandFoodResearch/hidecan/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../../logo.png" class="logo" alt=""><h1>Adding custom tracks</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/PlantandFoodResearch/hidecan/blob/main/vignettes/web_only/custom_tracks.Rmd" class="external-link"><code>vignettes/web_only/custom_tracks.Rmd</code></a></small>
      <div class="d-none name"><code>custom_tracks.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://plantandfoodresearch.github.io/hidecan/">hidecan</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://sjmgarnier.github.io/viridis/" class="external-link">viridis</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
<p>While hidecan was designed with the goal of representing GWAS and DE
results with lists of candidate genes, in principle it can be used to
visualise any genomic feature that can be mapped to a physical location
on the genome; for example significant QTLs, HiC data etc. To facilitate
this, it is possible to add one or more custom data types into a hidecan
plot.</p>
<div class="section level2">
<h2 id="adding-one-custom-data-type">Adding one custom data type<a class="anchor" aria-label="anchor" href="#adding-one-custom-data-type"></a>
</h2>
<p>Let us start by simulating a new set of results – for simplicity
we’ll assume that we have results from a QTL mapping on the same markers
that were used for the GWAS analysis:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/get_example_data.html">get_example_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">586</span><span class="op">)</span></span>
<span><span class="va">x</span><span class="op">$</span><span class="va">QTL</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">GWAS</span> <span class="op">|&gt;</span> </span>
<span>  <span class="co">## shuffling the scores around to get different "significant" markers</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">score</span>, <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>We can add this new data to the hidecan plot through the
<code>custom_list</code> argument of the <code><a href="../../reference/hidecan_plot.html">hidecan_plot()</a></code>
function (we’ll only look at two chromosomes for simplicity), and use
the <code>score_thr_custom</code> argument to set the significance
threshold on the score:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  chroms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ST4.03ch01"</span>, <span class="st">"ST4.03ch02"</span><span class="op">)</span>,</span>
<span>  custom_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"QTL"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  score_thr_custom <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="custom_tracks_files/figure-html/hidecan-plot-1.png" width="960"></p>
<p>Importantly, the data-frame that is passed to
<code>custom_list</code> must contain at least the following columns:
<code>chromosome</code>, <code>position</code> and <code>score</code>.
If the genomic features you want to add are genes for example (with a
start and end position) you need to calculate the <code>position</code>
column yourself.</p>
<p>We can customise the labels and colours used for this custom track
with the help of the <code><a href="../../reference/hidecan_aes.html">hidecan_aes()</a></code> function:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot_aes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/hidecan_aes.html">hidecan_aes</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">plot_aes</span>, max.level <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 4</span></span>
<span><span class="co">#&gt;  $ GWAS_data_thr  :List of 5</span></span>
<span><span class="co">#&gt;  $ DE_data_thr    :List of 5</span></span>
<span><span class="co">#&gt;  $ CAN_data_thr   :List of 5</span></span>
<span><span class="co">#&gt;  $ CUSTOM_data_thr:List of 5</span></span>
<span></span>
<span><span class="va">plot_aes</span><span class="op">[[</span><span class="st">"CUSTOM_data_thr"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; $y_label</span></span>
<span><span class="co">#&gt; [1] "Custom track"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $line_colour</span></span>
<span><span class="co">#&gt; [1] "darkgoldenrod2"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $point_shape</span></span>
<span><span class="co">#&gt; [1] 21</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $show_name</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $fill_scale</span></span>
<span><span class="co">#&gt; &lt;ScaleContinuous&gt;</span></span>
<span><span class="co">#&gt;  Range:  </span></span>
<span><span class="co">#&gt;  Limits:    0 --    1</span></span></code></pre></div>
<p>In particular, we will change the name given to the track, as well as
the name used for the legend:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot_aes</span><span class="op">$</span><span class="va">CUSTOM_data_thr</span><span class="op">$</span><span class="va">y_label</span> <span class="op">&lt;-</span> <span class="st">"Signif. QTLs"</span></span>
<span></span>
<span><span class="va">plot_aes</span><span class="op">$</span><span class="va">CUSTOM_data_thr</span><span class="op">$</span><span class="va">fill_scale</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="st">"QTL score"</span></span>
<span></span>
<span><span class="co"># We could also change the entire legend with:</span></span>
<span></span>
<span><span class="co"># default_aes$CUSTOM_data_thr$fill_scale &lt;- scale_fill_viridis(</span></span>
<span><span class="co">#   "QTL score",</span></span>
<span><span class="co">#   option = "inferno",</span></span>
<span><span class="co">#   guide = guide_colourbar(</span></span>
<span><span class="co">#     title.position = "top",</span></span>
<span><span class="co">#     title.hjust = 0.5,</span></span>
<span><span class="co">#     order = 4 # ensures this legend is shown after the one for DE scores</span></span>
<span><span class="co">#   )</span></span>
<span><span class="co"># )</span></span></code></pre></div>
<p>We will use these values in the hidecan plot by passing the list
through the <code>custom_aes</code> argument:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  chroms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ST4.03ch01"</span>, <span class="st">"ST4.03ch02"</span><span class="op">)</span>,</span>
<span>  custom_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"QTL"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  score_thr_custom <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  custom_aes <span class="op">=</span> <span class="va">plot_aes</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="custom_tracks_files/figure-html/hidecan-custom-aes-1.png" width="960"></p>
<p>Note that, as for the other supported data types, we can pass more
than one data-frame to <code>custom_list</code> to get several tracks of
QTLs:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Simulating a second set of QTL results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">779</span><span class="op">)</span></span>
<span><span class="va">x</span><span class="op">$</span><span class="va">QTL2</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">QTL</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">score</span>, <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,    </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,</span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  chroms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ST4.03ch01"</span>, <span class="st">"ST4.03ch02"</span><span class="op">)</span>,</span>
<span>  custom_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Trait 1"</span> <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"QTL"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                     <span class="st">"Trait 2"</span> <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"QTL2"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,</span>
<span>  score_thr_custom <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  custom_aes <span class="op">=</span> <span class="va">plot_aes</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="custom_tracks_files/figure-html/two-qtl-results-1.png" width="960"></p>
<p>By default, the values in the <code>score</code> column are shown as
the points colour, but we can hide it by setting the
<code>fill_scale</code> element of our aesthetics list to
<code>NULL</code>. Also, we can add labels to the significant points, by
adding a <code>name</code> column to the data-frame, and setting the
<code>show_name</code> element of our aesthetics list to
<code>TRUE</code>:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Creating fake names for the markers</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">342</span><span class="op">)</span></span>
<span><span class="va">x</span><span class="op">$</span><span class="va">QTL</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">QTL</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">LETTERS</span>, <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">plot_aes2</span> <span class="op">&lt;-</span> <span class="va">plot_aes</span></span>
<span></span>
<span><span class="va">plot_aes2</span><span class="op">$</span><span class="va">CUSTOM_data_thr</span><span class="op">$</span><span class="va">show_name</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">plot_aes2</span><span class="op">$</span><span class="va">CUSTOM_data_thr</span><span class="op">$</span><span class="va">fill_scale</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span></span>
<span><span class="fu"><a href="../../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  chroms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ST4.03ch01"</span>, <span class="st">"ST4.03ch02"</span><span class="op">)</span>,</span>
<span>  custom_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"QTL"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  score_thr_custom <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  custom_aes <span class="op">=</span> <span class="va">plot_aes2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="custom_tracks_files/figure-html/adding-names-1.png" width="960"></p>
</div>
<div class="section level2">
<h2 id="adding-several-custom-data-types">Adding several custom data types<a class="anchor" aria-label="anchor" href="#adding-several-custom-data-types"></a>
</h2>
<p>It is also possible to add more than one custom data type. For
example, let’s say that we want to add information about differentially
methylated regions to the plot. We’ll use the example DE data to
simulate our methylated dataset:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">554</span><span class="op">)</span></span>
<span><span class="va">x</span><span class="op">$</span><span class="va">METH</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">DE</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>    position <span class="op">=</span> <span class="op">(</span><span class="va">start</span> <span class="op">+</span> <span class="va">end</span><span class="op">)</span> <span class="op">/</span> <span class="fl">2</span>, <span class="co">## need to calculate the position of the regions</span></span>
<span>    score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">padj</span><span class="op">)</span>, <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>    name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">letters</span>, <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>Importantly, the significance threshold encoded by the
<code>score_thr_custom</code> argument of <code><a href="../../reference/hidecan_plot.html">hidecan_plot()</a></code>
will be used for all custom datasets, so if we don’t want to use the
same significance threshold on our QTL and methylation data, we need to
perform the filtering of significant features by hand before and then
set the threshold to 0:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span><span class="op">$</span><span class="va">QTL</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">QTL</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">score</span> <span class="op">&gt;=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">x</span><span class="op">$</span><span class="va">METH</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">METH</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">score</span> <span class="op">&gt;=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.04</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>The key to specifying that the two custom datasets are different data
types is to add to the new data-frame an argument called
<code>aes_type</code>, which is simply a label that tells the function
which aesthetics settings it should use from the list passsed to
<code>custom_aes</code>. We will retain the aesthetics we have set for
the QTL scores, and create new aesthetics for the methylation data:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Adding the aes_type attribute to the methylation data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/attr.html" class="external-link">attr</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">METH</span>, <span class="st">"aes_type"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"METH"</span></span>
<span></span>
<span><span class="co">## The name of the new element in the aesthetics list must match the value of</span></span>
<span><span class="co">## the aes_type attribute</span></span>
<span><span class="va">plot_aes</span><span class="op">$</span><span class="va">METH</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  y_label <span class="op">=</span> <span class="st">"DMRs"</span>,</span>
<span>  line_colour <span class="op">=</span> <span class="st">"purple"</span>,</span>
<span>  point_shape <span class="op">=</span> <span class="fl">24</span>,</span>
<span>  show_name <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  fill_scale <span class="op">=</span> <span class="fu"><a href="https://sjmgarnier.github.io/viridis/reference/scale_viridis.html" class="external-link">scale_fill_viridis</a></span><span class="op">(</span></span>
<span>    <span class="st">"DM score"</span>,</span>
<span>    option <span class="op">=</span> <span class="st">"mako"</span>,</span>
<span>    guide <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_colourbar.html" class="external-link">guide_colourbar</a></span><span class="op">(</span></span>
<span>      title.position <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>      title.hjust <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>      order <span class="op">=</span> <span class="fl">5</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  chroms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ST4.03ch01"</span>, <span class="st">"ST4.03ch02"</span><span class="op">)</span>,</span>
<span>  custom_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"QTL"</span><span class="op">]</span><span class="op">]</span>, <span class="va">x</span><span class="op">[[</span><span class="st">"METH"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,</span>
<span>  score_thr_custom <span class="op">=</span> <span class="fl">0</span>, <span class="co">## no filtering, has been done manually</span></span>
<span>  custom_aes <span class="op">=</span> <span class="va">plot_aes</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="custom_tracks_files/figure-html/unnamed-chunk-3-1.png" width="960"></p>
<p>There are no restrictions on the number of custom data types that can
be added to a hidecan plot.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Olivia Angelin-Bonnet.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
