<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Filters markers or genes/transcripts based on a threshold applied to their
GWAS or DE score, and log2(fold-change) (if applicable). For a set of
candidate genes, simply returns the list. Note that markers or genes with
a missing score or log2(fold-change) will be removed from the dataset."><title>Filters GWAS or DE results based on a threshold — apply_threshold • hidecan</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Filters GWAS or DE results based on a threshold — apply_threshold"><meta property="og:description" content="Filters markers or genes/transcripts based on a threshold applied to their
GWAS or DE score, and log2(fold-change) (if applicable). For a set of
candidate genes, simply returns the list. Note that markers or genes with
a missing score or log2(fold-change) will be removed from the dataset."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">hidecan</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/hidecan.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/hidecan-step-by-step.html">Step-by-step</a>
    <a class="dropdown-item" href="../articles/web_only/gwaspoly_output.html">GWASpoly output</a>
    <a class="dropdown-item" href="../articles/web_only/shiny-app.html">Shiny app</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/PlantandFoodResearch/hidecan/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Filters GWAS or DE results based on a threshold</h1>
      <small class="dont-index">Source: <a href="https://github.com/PlantandFoodResearch/hidecan/blob/HEAD/R/apply_threshold.R" class="external-link"><code>R/apply_threshold.R</code></a></small>
      <div class="d-none name"><code>apply_threshold.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Filters markers or genes/transcripts based on a threshold applied to their
GWAS or DE score, and log2(fold-change) (if applicable). For a set of
candidate genes, simply returns the list. Note that markers or genes with
a missing score or log2(fold-change) will be removed from the dataset.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">apply_threshold</span><span class="op">(</span><span class="va">x</span>, score_thr <span class="op">=</span> <span class="fl">0</span>, log2fc_thr <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for GWAS_data</span></span>
<span><span class="fu">apply_threshold</span><span class="op">(</span><span class="va">x</span>, score_thr <span class="op">=</span> <span class="fl">0</span>, log2fc_thr <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for DE_data</span></span>
<span><span class="fu">apply_threshold</span><span class="op">(</span><span class="va">x</span>, score_thr <span class="op">=</span> <span class="fl">0</span>, log2fc_thr <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for CAN_data</span></span>
<span><span class="fu">apply_threshold</span><span class="op">(</span><span class="va">x</span>, score_thr <span class="op">=</span> <span class="fl">0</span>, log2fc_thr <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for default</span></span>
<span><span class="fu">apply_threshold</span><span class="op">(</span><span class="va">x</span>, score_thr <span class="op">=</span> <span class="fl">0</span>, log2fc_thr <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>x</dt>
<dd><p>Either a <code>GWAS_data</code>, <code>DE_data</code> or <code>CAN_data</code> object.</p></dd>


<dt>score_thr</dt>
<dd><p>Numeric, threshold to use on markers' or genes/transcripts' score.
Only markers or genes with a score equal to or higher than this threshold
will be retained. Default value is 0. Ignored for <code>CAN_data</code>.</p></dd>


<dt>log2fc_thr</dt>
<dd><p>Numeric, threshold to use on the absolute value of genes/
transcripts' log2(fold-change). Only genes/transcripts with an absolute
log2(fold-change) equal to or higher than this threshold will be retained.
Ignored for <code>GWAS_data</code> and <code>CAN_data</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A filtered tibble (of class <code>GWAS_data_thr</code>, <code>DE_data_thr</code> or
<code>CAN_data_thr</code>).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="get_example_data.html">get_example_data</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## For GWAS results</span></span></span>
<span class="r-in"><span><span class="fu">apply_threshold</span><span class="op">(</span><span class="fu"><a href="GWAS_data.html">GWAS_data</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, score_thr <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 37 × 4</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    id                  chromosome position score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> ST4.03ch00_22680252 ST4.03ch00 22<span style="text-decoration: underline;">680</span>252  4.41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> ST4.03ch01_6317643  ST4.03ch01  6<span style="text-decoration: underline;">317</span>643  4.15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> ST4.03ch01_7671100  ST4.03ch01  7<span style="text-decoration: underline;">671</span>100  4.43</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> ST4.03ch01_8653747  ST4.03ch01  8<span style="text-decoration: underline;">653</span>747  4.69</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> ST4.03ch01_12842648 ST4.03ch01 12<span style="text-decoration: underline;">842</span>648  6.85</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> ST4.03ch01_13334335 ST4.03ch01 13<span style="text-decoration: underline;">334</span>335  5.24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> ST4.03ch01_13544786 ST4.03ch01 13<span style="text-decoration: underline;">544</span>786  4.00</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> ST4.03ch01_42743974 ST4.03ch01 42<span style="text-decoration: underline;">743</span>974  4.00</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> ST4.03ch01_42949112 ST4.03ch01 42<span style="text-decoration: underline;">949</span>112  4.18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> ST4.03ch01_52529264 ST4.03ch01 52<span style="text-decoration: underline;">529</span>264  4.85</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 27 more rows</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## For DE results - in second line, no threshold is applied</span></span></span>
<span class="r-in"><span><span class="co">## on the log2(fold-change)</span></span></span>
<span class="r-in"><span><span class="fu">apply_threshold</span><span class="op">(</span><span class="fu"><a href="DE_data.html">DE_data</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, score_thr <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>, log2fc_thr <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 5 × 9</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   gene       chromosome   padj log2FoldChange  start    end label score position</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> PGSC0003D… ST4.03ch01 0.048<span style="text-decoration: underline;">8</span>           1.61 8.15<span style="color: #949494;">e</span>7 8.15<span style="color: #949494;">e</span>7 Gene…  1.31   8.15<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> PGSC0003D… ST4.03ch03 0.048<span style="text-decoration: underline;">8</span>           1.14 3.49<span style="color: #949494;">e</span>7 3.49<span style="color: #949494;">e</span>7 Gene…  1.31   3.49<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> PGSC0003D… ST4.03ch03 0.018<span style="text-decoration: underline;">4</span>           3.06 5.99<span style="color: #949494;">e</span>7 5.99<span style="color: #949494;">e</span>7 Phos…  1.73   5.99<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> PGSC0003D… ST4.03ch05 0.041<span style="text-decoration: underline;">4</span>           1.10 3.47<span style="color: #949494;">e</span>6 3.47<span style="color: #949494;">e</span>6 Cons…  1.38   3.47<span style="color: #949494;">e</span>6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> PGSC0003D… ST4.03ch09 0.032<span style="text-decoration: underline;">8</span>           1.51 5.40<span style="color: #949494;">e</span>7 5.40<span style="color: #949494;">e</span>7 Vici…  1.48   5.40<span style="color: #949494;">e</span>7</span>
<span class="r-in"><span><span class="fu">apply_threshold</span><span class="op">(</span><span class="fu"><a href="DE_data.html">DE_data</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, score_thr <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>, log2fc_thr <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 57 × 9</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    gene     chromosome    padj log2FoldChange  start    end label score position</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> PGSC000… ST4.03ch00 0.048<span style="text-decoration: underline;">8</span>           0.838 3.80<span style="color: #949494;">e</span>7 3.80<span style="color: #949494;">e</span>7 Pept…  1.31   3.80<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> PGSC000… ST4.03ch01 0.036<span style="text-decoration: underline;">2</span>          -<span style="color: #BB0000;">0.181</span> 5.83<span style="color: #949494;">e</span>7 5.83<span style="color: #949494;">e</span>7 Cons…  1.44   5.83<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> PGSC000… ST4.03ch01 0.048<span style="text-decoration: underline;">8</span>           1.61  8.15<span style="color: #949494;">e</span>7 8.15<span style="color: #949494;">e</span>7 Gene…  1.31   8.15<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> PGSC000… ST4.03ch01 0.004<span style="text-decoration: underline;">65</span>          0.438 8.45<span style="color: #949494;">e</span>7 8.45<span style="color: #949494;">e</span>7 Phi-2  2.33   8.45<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> PGSC000… ST4.03ch02 0.048<span style="text-decoration: underline;">8</span>           0.325 2.26<span style="color: #949494;">e</span>7 2.26<span style="color: #949494;">e</span>7 Cons…  1.31   2.26<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> PGSC000… ST4.03ch02 0.026<span style="text-decoration: underline;">6</span>           0.768 3.32<span style="color: #949494;">e</span>7 3.32<span style="color: #949494;">e</span>7 Leuc…  1.58   3.32<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> PGSC000… ST4.03ch02 0.048<span style="text-decoration: underline;">8</span>           0.381 4.14<span style="color: #949494;">e</span>7 4.14<span style="color: #949494;">e</span>7 Cons…  1.31   4.14<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> PGSC000… ST4.03ch02 0.031<span style="text-decoration: underline;">4</span>           0.663 4.23<span style="color: #949494;">e</span>7 4.23<span style="color: #949494;">e</span>7 Home…  1.50   4.23<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> PGSC000… ST4.03ch03 0.048<span style="text-decoration: underline;">8</span>          -<span style="color: #BB0000;">0.804</span> 5.33<span style="color: #949494;">e</span>5 5.41<span style="color: #949494;">e</span>5 SNF2…  1.31   5.37<span style="color: #949494;">e</span>5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> PGSC000… ST4.03ch03 0.049<span style="text-decoration: underline;">3</span>          -<span style="color: #BB0000;">0.345</span> 9.26<span style="color: #949494;">e</span>6 9.26<span style="color: #949494;">e</span>6 Dihy…  1.31   9.26<span style="color: #949494;">e</span>6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 47 more rows</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## No effect on the Candidate genes</span></span></span>
<span class="r-in"><span><span class="fu">apply_threshold</span><span class="op">(</span><span class="fu"><a href="CAN_data.html">CAN_data</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 32 × 7</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    id                    chromosome    start      end name    gene_name position</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> PGSC0003DMG400003155  ST4.03ch03 46<span style="text-decoration: underline;">757</span>152 46<span style="text-decoration: underline;">762</span>127 4CL     4-coumar…   4.68<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> PGSC0003DMG400014223  ST4.03ch03 57<span style="text-decoration: underline;">466</span>692 57<span style="text-decoration: underline;">469</span>946 4CL2    4-coumar…   5.75<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> PGSC0003DMG400011189  ST4.03ch07  1<span style="text-decoration: underline;">001</span>854  1<span style="text-decoration: underline;">006</span>278 HQT     HQT         1.00<span style="color: #949494;">e</span>6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> PGSC0003DMG400005492  ST4.03ch05 36<span style="text-decoration: underline;">342</span>746 36<span style="text-decoration: underline;">347</span>409 PAL     phenylal…   3.63<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> PGSC0003DMG400005279  ST4.03ch05 42<span style="text-decoration: underline;">523</span>943 42<span style="text-decoration: underline;">525</span>912 peroxi… peroxida…   4.25<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> PGSC0003DMG400007782  ST4.03ch03 38<span style="text-decoration: underline;">537</span>202 38<span style="text-decoration: underline;">540</span>209 PHO1A   PHO1A       3.85<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> PGSC0003DMG400023182  ST4.03ch02 10<span style="text-decoration: underline;">091</span>564 10<span style="text-decoration: underline;">093</span>164 LIPIII… LipIII27    1.01<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> PGSC0003DMG400029576  ST4.03ch08 45<span style="text-decoration: underline;">676</span>042 45<span style="text-decoration: underline;">677</span>832 PPOs    Polyphen…   4.57<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> PGSC0003DMG400018924  ST4.03ch08 45<span style="text-decoration: underline;">778</span>567 45<span style="text-decoration: underline;">780</span>333 <span style="color: #BB0000;">NA</span>      Polyphen…   4.58<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> PGSC0003DMG400018919  ST4.03ch08 45<span style="text-decoration: underline;">793</span>661 45<span style="text-decoration: underline;">794</span>229 <span style="color: #BB0000;">NA</span>      Polyphen…   4.58<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 22 more rows</span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Olivia Angelin-Bonnet.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

