<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Wrapper to create a HIDECAN plot — hidecan_plot • hidecan</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Wrapper to create a HIDECAN plot — hidecan_plot"><meta name="description" content="Wrapper function to create a HIDECAN plot from GWAS results, DE results or
candidate genes."><meta property="og:description" content="Wrapper function to create a HIDECAN plot from GWAS results, DE results or
candidate genes."><meta property="og:image" content="https://plantandfoodresearch.github.io/hidecan/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hidecan</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/hidecan.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/hidecan-step-by-step.html">Step-by-step</a></li>
    <li><a class="dropdown-item" href="../articles/web_only/gwaspoly_output.html">GWASpoly output</a></li>
    <li><a class="dropdown-item" href="../articles/web_only/custom_tracks.html">Adding custom tracks</a></li>
    <li><a class="dropdown-item" href="../articles/web_only/shiny-app.html">Shiny app</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/PlantandFoodResearch/hidecan/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Wrapper to create a HIDECAN plot</h1>
      <small class="dont-index">Source: <a href="https://github.com/PlantandFoodResearch/hidecan/blob/main/R/create_hidecan_plot.R" class="external-link"><code>R/create_hidecan_plot.R</code></a></small>
      <div class="d-none name"><code>hidecan_plot.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Wrapper function to create a HIDECAN plot from GWAS results, DE results or
candidate genes.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">hidecan_plot</span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  de_list <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  can_list <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  score_thr_gwas <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  score_thr_de <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  chrom_length <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  colour_genes_by_score <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  remove_empty_chrom <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  chroms <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  chrom_limits <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  title <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  subtitle <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  n_rows <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  n_cols <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  legend_position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">3.5</span>,</span>
<span>  label_padding <span class="op">=</span> <span class="fl">0.15</span>,</span>
<span>  custom_aes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  custom_list <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  score_thr_custom <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-gwas-list">gwas_list<a class="anchor" aria-label="anchor" href="#arg-gwas-list"></a></dt>
<dd><p>Data-frame or list of data-frames containing GWAS results,
each with at least a <code>chromosome</code>, <code>position</code> and either <code>padj</code> or <code>score</code>
columns. If a named list, the names will be used in the plot.</p></dd>


<dt id="arg-de-list">de_list<a class="anchor" aria-label="anchor" href="#arg-de-list"></a></dt>
<dd><p>Data-frame or list of data-frames containing DE results, each
with at least a <code>chromosome</code>, <code>start</code>, <code>end</code>, <code>log2FoldChange</code> and either
<code>padj</code> or <code>score</code> columns. If a named list, the names will be used in the
plot.</p></dd>


<dt id="arg-can-list">can_list<a class="anchor" aria-label="anchor" href="#arg-can-list"></a></dt>
<dd><p>Data-frame or list of data-frames containing candidate genes,
each with at least a <code>chromosome</code>, <code>start</code>, <code>end</code> and <code>name</code> columns. If a
named list, the names will be used in the plot.</p></dd>


<dt id="arg-score-thr-gwas">score_thr_gwas<a class="anchor" aria-label="anchor" href="#arg-score-thr-gwas"></a></dt>
<dd><p>Numeric, the score threshold for GWAS results that will
be used to select which markers will be plotted. Default value is 4.</p></dd>


<dt id="arg-score-thr-de">score_thr_de<a class="anchor" aria-label="anchor" href="#arg-score-thr-de"></a></dt>
<dd><p>Numeric, the score threshold for DE results that will be
used to select which markers will be plotted. Default value is 2.</p></dd>


<dt id="arg-log-fc-thr">log2fc_thr<a class="anchor" aria-label="anchor" href="#arg-log-fc-thr"></a></dt>
<dd><p>Numeric, the log2(fold-change) threshold that will be used
to select which genes will be plotted. Default value is 1.</p></dd>


<dt id="arg-chrom-length">chrom_length<a class="anchor" aria-label="anchor" href="#arg-chrom-length"></a></dt>
<dd><p>Optional, tibble with columns <code>chromosome</code> and <code>length</code>,
giving for each chromosome its length in bp. If <code>NULL</code> (the default), will
be inferred from the GWAS, DE and candidate gene data.</p></dd>


<dt id="arg-colour-genes-by-score">colour_genes_by_score<a class="anchor" aria-label="anchor" href="#arg-colour-genes-by-score"></a></dt>
<dd><p>Logical, whether to colour the genes by score
(<code>TRUE</code>) or by log2(fold-change) (<code>FALSE</code>). Default value is <code>TRUE</code>.</p></dd>


<dt id="arg-remove-empty-chrom">remove_empty_chrom<a class="anchor" aria-label="anchor" href="#arg-remove-empty-chrom"></a></dt>
<dd><p>Logical, should chromosomes with no significant
markers/genes nor candidate genes be removed from the plot? Default value
if <code>FALSE</code>.</p></dd>


<dt id="arg-chroms">chroms<a class="anchor" aria-label="anchor" href="#arg-chroms"></a></dt>
<dd><p>Character vector, name of chromosomes to include in the plot.
If <code>NULL</code> (default value), all chromosomes will be included.</p></dd>


<dt id="arg-chrom-limits">chrom_limits<a class="anchor" aria-label="anchor" href="#arg-chrom-limits"></a></dt>
<dd><p>Integer vector of length 2, or named list where the
elements are integer vectors of length 2. If vector, gives the lower and upper
limit of the chromosomes (in bp) to use in the plot. If a named list, names
should correspond to chromosome names. Gives for each chromosome the lower
and upper limits (in bp) to use in the plot. Doesn't have to be specified
for all chromosomes. Default value is <code>NULL</code>, i.e. no limits are applied
to the chromosomes (they will be plotted in their entirety).</p></dd>


<dt id="arg-title">title<a class="anchor" aria-label="anchor" href="#arg-title"></a></dt>
<dd><p>Character, title of the plot. Default value is <code>NULL</code> (i.e.
no title will be added to the plot).</p></dd>


<dt id="arg-subtitle">subtitle<a class="anchor" aria-label="anchor" href="#arg-subtitle"></a></dt>
<dd><p>Character, subtitle of the plot. Default value is <code>NULL</code>
(i.e. no subtitle will be added to the plot).</p></dd>


<dt id="arg-n-rows">n_rows<a class="anchor" aria-label="anchor" href="#arg-n-rows"></a></dt>
<dd><p>Integer, number of rows of chromosomes to create in the plot.
Default value is <code>NULL</code>.</p></dd>


<dt id="arg-n-cols">n_cols<a class="anchor" aria-label="anchor" href="#arg-n-cols"></a></dt>
<dd><p>Integer, number of columns of chromosomes to create in the plot.
Default value is 2. Will be set to <code>NULL</code> if <code>n_rows</code> is not <code>NULL</code>.</p></dd>


<dt id="arg-legend-position">legend_position<a class="anchor" aria-label="anchor" href="#arg-legend-position"></a></dt>
<dd><p>Character, position of the legend in the plot. Can be
<code>bottom</code> (default value), <code>top</code>, <code>right</code>, <code>left</code> or <code>none</code>.</p></dd>


<dt id="arg-point-size">point_size<a class="anchor" aria-label="anchor" href="#arg-point-size"></a></dt>
<dd><p>Numeric, size of the points in the plot. Default value is 3.</p></dd>


<dt id="arg-label-size">label_size<a class="anchor" aria-label="anchor" href="#arg-label-size"></a></dt>
<dd><p>Numeric, size of the gene labels in the plot. Default value is
3.5 (for <code><a href="https://ggrepel.slowkow.com/reference/geom_text_repel.html" class="external-link">geom_label_repel</a></code>).</p></dd>


<dt id="arg-label-padding">label_padding<a class="anchor" aria-label="anchor" href="#arg-label-padding"></a></dt>
<dd><p>Numeric, amount of padding around gene labels in the plot,
as unit or number. Default value is 0.15
(for <a href="https://ggrepel.slowkow.com/reference/geom_text_repel.html" class="external-link">geom_label_repel</a>).</p></dd>


<dt id="arg-custom-aes">custom_aes<a class="anchor" aria-label="anchor" href="#arg-custom-aes"></a></dt>
<dd><p>Named list of plot aesthetics for custom data types. See
<code><a href="hidecan_aes.html">hidecan_aes()</a></code> for information about the content of each element. Default
is <code>NULL</code> (only needed if there are <code>CUSTOM_data_thr</code> objects in <code>x</code> or to
customise the aesthetics for the default data tracks).</p></dd>


<dt id="arg-custom-list">custom_list<a class="anchor" aria-label="anchor" href="#arg-custom-list"></a></dt>
<dd><p>Data-frame or list of data-frames containing custom
genomic features, each with at least a <code>chromosome</code>, <code>position</code> and <code>score</code>
columns. If a named list, the names will be used in the plot.</p></dd>


<dt id="arg-score-thr-custom">score_thr_custom<a class="anchor" aria-label="anchor" href="#arg-score-thr-custom"></a></dt>
<dd><p>Numeric, the score threshold for custom genomic
features that will be used to select which markers will be plotted. Default
value is 0.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>a <code>ggplot</code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="get_example_data.html">get_example_data</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Typical example with one GWAs result table, one DE result table and</span></span></span>
<span class="r-in"><span><span class="co">## one table of candidate genes</span></span></span>
<span class="r-in"><span><span class="fu">hidecan_plot</span><span class="op">(</span>gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>             de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>             can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>             score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>             score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.005</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>             log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span></span>
<span class="r-in"><span>             label_size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Example with two sets of GWAS results</span></span></span>
<span class="r-in"><span><span class="fu">hidecan_plot</span><span class="op">(</span>gwas_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>, <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>             score_thr_gwas <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Example with two sets of DE results, with names</span></span></span>
<span class="r-in"><span><span class="fu">hidecan_plot</span><span class="op">(</span>de_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"X vs Y"</span> <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>                            <span class="st">"X vs Z"</span> <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>             score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>             log2fc_thr <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Set limits on all chromosomes (to "zoom in" to the 10-20Mb region)</span></span></span>
<span class="r-in"><span><span class="fu">hidecan_plot</span><span class="op">(</span>gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>             de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>             can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>             score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>             score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.005</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>             log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span></span>
<span class="r-in"><span>             label_size <span class="op">=</span> <span class="fl">2</span>,</span></span>
<span class="r-in"><span>             chrom_limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10e6</span>, <span class="fl">20e6</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Set limits on some chromosomes only</span></span></span>
<span class="r-in"><span><span class="fu">hidecan_plot</span><span class="op">(</span>gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>             de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>             can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>             score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>             score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.005</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>             log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span></span>
<span class="r-in"><span>             label_size <span class="op">=</span> <span class="fl">2</span>,</span></span>
<span class="r-in"><span>             chrom_limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"ST4.03ch00"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10e6</span>, <span class="fl">20e6</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                                  <span class="st">"ST4.03ch02"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">15e6</span>, <span class="fl">25e6</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Olivia Angelin-Bonnet.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

