<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="hidecan">
<title>HIDECAN plot from GWASpoly output â€¢ hidecan</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="HIDECAN plot from GWASpoly output">
<meta property="og:description" content="hidecan">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">hidecan</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/hidecan.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/hidecan-step-by-step.html">Step-by-step</a>
    <a class="dropdown-item" href="../articles/gwaspoly_output.html">GWASpoly output</a>
    <a class="dropdown-item" href="../articles/shiny-app.html">Shiny app</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/PlantandFoodResearch/hidecan/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>HIDECAN plot from GWASpoly output</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/PlantandFoodResearch/hidecan/blob/HEAD/vignettes/gwaspoly_output.Rmd" class="external-link"><code>vignettes/gwaspoly_output.Rmd</code></a></small>
      <div class="d-none name"><code>gwaspoly_output.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://shiny-adventure-acfe02d1.pages.github.io/">hidecan</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GWASpoly</span><span class="op">)</span></span></code></pre></div>
<p>The <a href="https://github.com/jendelman/GWASpoly" class="external-link">GWASpoly</a>
package is designed to perform GWAS analyses for autopolyploid
organisms. It allows the simultaneous analysis of several traits or
phenotypes, and can compute markers association scores with different
genetic models. The package also includes a function to compute the
significance threshold for each trait and genetic model.</p>
<p>The <code>hidecan</code> package provides functions to generate a
HIDECAN plot directly from a <code>GWASpoly</code> output object (object
of class <code>GWASpoly.thresh</code> obtained with the
<code><a href="https://rdrr.io/pkg/GWASpoly/man/set.threshold.html" class="external-link">GWASpoly::set.threshold()</a></code>).</p>
<div class="section level2">
<h2 id="gwaspoly-example-data">GWASpoly example data<a class="anchor" aria-label="anchor" href="#gwaspoly-example-data"></a>
</h2>
<p>An example of GWASpoly output, based on the <a href="https://acsess.onlinelibrary.wiley.com/doi/10.3835/plantgenome2015.08.0073" class="external-link">original
example dataset</a> from the GWASpoly package, can be obtained with the
<code><a href="../reference/get_gwaspoly_example_data.html">get_gwaspoly_example_data()</a></code> function:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwaspoly_output</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_gwaspoly_example_data.html">get_gwaspoly_example_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">gwaspoly_output</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "GWASpoly.thresh"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "GWASpoly"</span></span></code></pre></div>
<p>In this example, three traits were analysed:
<code>tuber_eye_depth</code>, <code>tuber_shape</code> and
<code>sucrose</code>. For each trait, GWAS scores were computed with
four different genetic models: <code>general</code>,
<code>additive</code>, <code>1-dom-alt</code> and
<code>1-dom-ref</code>:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Traits analysed</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gwaspoly_output</span><span class="op">@</span><span class="va">scores</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "tuber_eye_depth" "tuber_shape"     "sucrose"</span></span>
<span></span>
<span><span class="co">## Genetic models tests</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gwaspoly_output</span><span class="op">@</span><span class="va">scores</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;             general  additive 1-dom-alt 1-dom-ref</span></span>
<span><span class="co">#&gt; c2_41437 0.45584445 0.4774282 0.1430731        NA</span></span>
<span><span class="co">#&gt; c2_24258 0.32552500 0.1655215        NA 0.0431065</span></span>
<span><span class="co">#&gt; c2_21332 0.06251865 0.2263273 0.1562208 0.4488712</span></span>
<span><span class="co">#&gt; c2_21320 0.92927679 0.5774697        NA 1.0170807</span></span>
<span><span class="co">#&gt; c2_21318 0.09043940 0.2200563 0.1410284        NA</span></span>
<span><span class="co">#&gt; c2_21314 0.19480464 0.6442068        NA 0.4286548</span></span></code></pre></div>
<p>See the Appendix section at the bottom of this vignette for the code
used to generate this example data.</p>
</div>
<div class="section level2">
<h2 id="hidecan-plot-from-gwaspoly-output">HIDECAN plot from GWASpoly output<a class="anchor" aria-label="anchor" href="#hidecan-plot-from-gwaspoly-output"></a>
</h2>
<p>The <code><a href="../reference/hidecan_plot_from_gwaspoly.html">hidecan_plot_from_gwaspoly()</a></code> function reads in a
<code>GWASpoly.thresh</code> object, extracts from it the marker scores
for each combination of trait and genetic model, and uses them to
construct a HIDECAN plot. In the y-axis, the trait is indicated first,
and the genetic model next in brackets:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot_from_gwaspoly.html">hidecan_plot_from_gwaspoly</a></span><span class="op">(</span></span>
<span>  <span class="va">gwaspoly_output</span>,</span>
<span>  remove_empty_chrom <span class="op">=</span> <span class="cn">TRUE</span>  </span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="gwaspoly_output_files/figure-html/hidecan-plot-from-gwaspoly-1.png" width="768"></p>
<p>From the HIDECAN plot, we can easily see that there is a genomic
region around 50Mb on chromosome 10 that is significantly associated
with both tuber eye depth and shape, when using either the general or
additive model. This region is not significantly associated with either
of these traits when considering one of the simplex dominant models. For
the sucrose phenotype, only the general model detected any significant
markers.</p>
<p>It is possible to specify which traits and/or genetic models are
represented in the HIDECAN plot, via the <code>traits</code> and
<code>models</code> arguments:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot_from_gwaspoly.html">hidecan_plot_from_gwaspoly</a></span><span class="op">(</span></span>
<span>  <span class="va">gwaspoly_output</span>,</span>
<span>  traits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"tuber_eye_depth"</span>, <span class="st">"tuber_shape"</span><span class="op">)</span>,</span>
<span>  models <span class="op">=</span> <span class="st">"general"</span>,</span>
<span>  remove_empty_chrom <span class="op">=</span> <span class="cn">TRUE</span>  </span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="gwaspoly_output_files/figure-html/selecting-traits-and-models-1.png" width="768"></p>
</div>
<div class="section level2">
<h2 id="the-gwaspoly-constructor">The GWASpoly constructor<a class="anchor" aria-label="anchor" href="#the-gwaspoly-constructor"></a>
</h2>
<p>Under the hood, the <code><a href="../reference/hidecan_plot_from_gwaspoly.html">hidecan_plot_from_gwaspoly()</a></code>
function relies on the <code><a href="../reference/GWAS_data_from_gwaspoly.html">GWAS_data_from_gwaspoly()</a></code>
constructor, which takes as an input either:</p>
<ul>
<li><p>a <code>GWASpoly.fitted</code> object (returned by the
<code><a href="https://rdrr.io/pkg/GWASpoly/man/GWASpoly.html" class="external-link">GWASpoly::GWASpoly()</a></code> function), or</p></li>
<li><p>a <code>GWASpoly.thresh</code> object (returned by the
<code><a href="https://rdrr.io/pkg/GWASpoly/man/set.threshold.html" class="external-link">GWASpoly::set.threshold()</a></code> function).</p></li>
</ul>
<p>The function extracts the marker scores for all traits and genetic
models present in the GWASpoly output, as well as the length of all
chromosomes. In addition, if the input data is a
<code>GWASpoly.thresh</code> object, it extracts the significance
threshold for each combination of trait and genetic model, and uses it
to filter significant markers.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwaspoly_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWAS_data_from_gwaspoly.html">GWAS_data_from_gwaspoly</a></span><span class="op">(</span><span class="va">gwaspoly_output</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## GWAS_data objects, i.e. tibbles of marker scores</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">gwaspoly_data</span><span class="op">$</span><span class="va">gwas_data_list</span>, max.level <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 12</span></span>
<span><span class="co">#&gt;  $ tuber_eye_depth (general)  : GWAS_dat [3,507 Ã— 4] (S3: GWAS_data/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ tuber_eye_depth (additive) : GWAS_dat [3,507 Ã— 4] (S3: GWAS_data/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ tuber_eye_depth (1-dom-alt): GWAS_dat [2,054 Ã— 4] (S3: GWAS_data/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ tuber_eye_depth (1-dom-ref): GWAS_dat [2,174 Ã— 4] (S3: GWAS_data/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ tuber_shape (general)      : GWAS_dat [3,507 Ã— 4] (S3: GWAS_data/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ tuber_shape (additive)     : GWAS_dat [3,507 Ã— 4] (S3: GWAS_data/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ tuber_shape (1-dom-alt)    : GWAS_dat [2,054 Ã— 4] (S3: GWAS_data/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ tuber_shape (1-dom-ref)    : GWAS_dat [2,174 Ã— 4] (S3: GWAS_data/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ sucrose (general)          : GWAS_dat [3,506 Ã— 4] (S3: GWAS_data/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ sucrose (additive)         : GWAS_dat [3,507 Ã— 4] (S3: GWAS_data/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ sucrose (1-dom-alt)        : GWAS_dat [2,054 Ã— 4] (S3: GWAS_data/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ sucrose (1-dom-ref)        : GWAS_dat [2,174 Ã— 4] (S3: GWAS_data/tbl_df/tbl/data.frame)</span></span>
<span></span>
<span><span class="co">## GWAS_data_thr objects, i.e. tibbles of significant markers</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">gwaspoly_data</span><span class="op">$</span><span class="va">gwas_data_thr_list</span>, max.level <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 12</span></span>
<span><span class="co">#&gt;  $ tuber_eye_depth (general)  : GWAS_dt_ [1 Ã— 4] (S3: GWAS_data_thr/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ tuber_eye_depth (additive) : GWAS_dt_ [3 Ã— 4] (S3: GWAS_data_thr/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ tuber_eye_depth (1-dom-alt): GWAS_dt_ [0 Ã— 4] (S3: GWAS_data_thr/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ tuber_eye_depth (1-dom-ref): GWAS_dt_ [0 Ã— 4] (S3: GWAS_data_thr/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ tuber_shape (general)      : GWAS_dt_ [2 Ã— 4] (S3: GWAS_data_thr/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ tuber_shape (additive)     : GWAS_dt_ [2 Ã— 4] (S3: GWAS_data_thr/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ tuber_shape (1-dom-alt)    : GWAS_dt_ [0 Ã— 4] (S3: GWAS_data_thr/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ tuber_shape (1-dom-ref)    : GWAS_dt_ [0 Ã— 4] (S3: GWAS_data_thr/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ sucrose (general)          : GWAS_dt_ [2 Ã— 4] (S3: GWAS_data_thr/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ sucrose (additive)         : GWAS_dt_ [0 Ã— 4] (S3: GWAS_data_thr/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ sucrose (1-dom-alt)        : GWAS_dt_ [0 Ã— 4] (S3: GWAS_data_thr/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ sucrose (1-dom-ref)        : GWAS_dt_ [0 Ã— 4] (S3: GWAS_data_thr/tbl_df/tbl/data.frame)</span></span>
<span></span>
<span><span class="co">## Chromosomes length</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">gwaspoly_data</span><span class="op">$</span><span class="va">chrom_length</span><span class="op">)</span></span>
<span><span class="co">#&gt; tibble [13 Ã— 2] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ chromosome: Ord.factor w/ 13 levels "0"&lt;"1"&lt;"2"&lt;"3"&lt;..: 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">#&gt;  $ length    : int [1:13] 36454137 88583876 48564909 61870684 72026885 51998374 59263222 56628128 56785385 61466245 ...</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="appendix-reproducing-the-gwaspoly-example-data">Appendix: reproducing the GWASpoly example data<a class="anchor" aria-label="anchor" href="#appendix-reproducing-the-gwaspoly-example-data"></a>
</h2>
<p>The example dataset provided through
<code><a href="../reference/get_gwaspoly_example_data.html">get_gwaspoly_example_data()</a></code> can be reproduced with the
following code:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GWASpoly</span><span class="op">)</span></span>
<span></span>
<span><span class="va">genofile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TableS1.csv"</span>, package <span class="op">=</span> <span class="st">"GWASpoly"</span><span class="op">)</span></span>
<span><span class="va">phenofile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TableS2.csv"</span>, package <span class="op">=</span> <span class="st">"GWASpoly"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Reading example data</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GWASpoly/man/read.GWASpoly.html" class="external-link">read.GWASpoly</a></span><span class="op">(</span></span>
<span>  ploidy <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  pheno.file <span class="op">=</span> <span class="va">phenofile</span>,</span>
<span>  geno.file <span class="op">=</span> <span class="va">genofile</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"ACGT"</span>,</span>
<span>  n.traits <span class="op">=</span> <span class="fl">13</span>,</span>
<span>  delim <span class="op">=</span> <span class="st">","</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Computing K matrix</span></span>
<span><span class="va">data.original</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GWASpoly/man/set.K.html" class="external-link">set.K</a></span><span class="op">(</span></span>
<span>  <span class="va">data</span>,</span>
<span>  LOCO <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  n.core <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Performing GWAS</span></span>
<span><span class="va">gwaspoly_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GWASpoly/man/GWASpoly.html" class="external-link">GWASpoly</a></span><span class="op">(</span></span>
<span>  <span class="va">data.original</span>,</span>
<span>  models <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"general"</span>, <span class="st">"additive"</span>, <span class="st">"1-dom"</span><span class="op">)</span>,</span>
<span>  traits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"tuber_eye_depth"</span>, <span class="st">"tuber_shape"</span>, <span class="st">"sucrose"</span><span class="op">)</span>,</span>
<span>  n.core <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Computing significance threshold</span></span>
<span><span class="co">## Object returned by get_gwaspoly_example_data()</span></span>
<span><span class="va">gwaspoly_res_thr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GWASpoly/man/set.threshold.html" class="external-link">set.threshold</a></span><span class="op">(</span></span>
<span>  <span class="va">gwaspoly_res</span>, </span>
<span>  method <span class="op">=</span> <span class="st">"M.eff"</span>, </span>
<span>  level <span class="op">=</span> <span class="fl">0.05</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Olivia Angelin-Bonnet.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
