<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="hidecan">
<title>hidecan • hidecan</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="hidecan">
<meta property="og:description" content="hidecan">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">hidecan</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/hidecan.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/hidecan-step-by-step.html">Step-by-step</a>
    <a class="dropdown-item" href="../articles/gwaspoly_output.html">GWASpoly output</a>
    <a class="dropdown-item" href="../articles/shiny-app.html">Shiny app</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/PlantandFoodResearch/hidecan/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>hidecan</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/PlantandFoodResearch/hidecan/blob/HEAD/vignettes/hidecan.Rmd" class="external-link"><code>vignettes/hidecan.Rmd</code></a></small>
      <div class="d-none name"><code>hidecan.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://shiny-adventure-acfe02d1.pages.github.io/" class="external-link">hidecan</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="input-data">Input data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h2>
<p>The <code>hidecan</code> package takes as input tibbles (data-frames)
of GWAS and DE results or candidate genes. The input data-frames should
contain some mandatory columns, depending on the type of data.</p>
<p>A list of example input datasets can be obtained via the
<code><a href="../reference/get_example_data.html">get_example_data()</a></code> function:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_example_data.html">get_example_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">x</span>, max.level <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 3</span></span>
<span><span class="co">#&gt;  $ GWAS: tibble [72,847 × 4] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ DE  : tibble [25,163 × 7] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ CAN : tibble [32 × 6] (S3: tbl_df/tbl/data.frame)</span></span></code></pre></div>
<div class="section level3">
<h3 id="gwas-results">GWAS results<a class="anchor" aria-label="anchor" href="#gwas-results"></a>
</h3>
<p>GWAS results should be provided as a tibble or data-frame, with one
row per genetic marker. The data-frame should contain at least the
following columns:</p>
<ul>
<li><p><code>chromosome</code>: character column, giving the ID of the
chromosome on which each marker is located;</p></li>
<li><p><code>position</code>: numeric column, the physical position
along the chromosome (in base pairs - bp) of the marker;</p></li>
<li><p>either <code>score</code> or <code>padj</code>: numeric column,
providing either the score (i.e. <code>-log10(p-value)</code>) or the
adjusted p-value of the marker. If a <code>score</code> column is
provided, the <code>padj</code> column will be ignored. If only a
<code>padj</code> column is provided, a <code>score</code> column will
be constructed as <code>-log10(padj)</code>.</p></li>
</ul>
<p>Any other column present in the data-frame will be ignored. An
example of valid input is shown below:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;   id                chromosome position  score</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> ST4.03ch00_172333 ST4.03ch00   <span style="text-decoration: underline;">172</span>333 0.342 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> ST4.03ch00_172690 ST4.03ch00   <span style="text-decoration: underline;">172</span>690 0.179 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> ST4.03ch00_194668 ST4.03ch00   <span style="text-decoration: underline;">194</span>668 0.047<span style="text-decoration: underline;">9</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> ST4.03ch00_194695 ST4.03ch00   <span style="text-decoration: underline;">194</span>695 0.469 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> ST4.03ch00_194712 ST4.03ch00   <span style="text-decoration: underline;">194</span>712 0.893 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> ST4.03ch00_275624 ST4.03ch00   <span style="text-decoration: underline;">275</span>624 0.380</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="differential-expression-results">Differential expression results<a class="anchor" aria-label="anchor" href="#differential-expression-results"></a>
</h3>
<p>DE results should be provided as a tibble or data-frame, with one row
per gene. The data-frame should contain at least the following
columns:</p>
<ul>
<li><p><code>chromosome</code>: character column, giving the ID of the
chromosome on which each gene is located;</p></li>
<li><p><code>start</code> and <code>end</code>: numeric columns, giving
the starting and end position of the gene along the chromosome (in bp).
These two columns will be used to calculate the position of the gene as
the half-way point between the start and end of the gene.</p></li>
<li><p>either <code>score</code> or <code>padj</code>: numeric column,
providing either the score (i.e. <code>-log10(p-value)</code>) or the
adjusted p-value of the gene. If a <code>score</code> column is
provided, the <code>padj</code> column will be ignored. If only a
<code>padj</code> column is provided, a <code>score</code> column will
be constructed as <code>-log10(padj)</code>.</p></li>
<li><p>either <code>foldChange</code> or <code>log2FoldChange</code>:
numeric column, giving either the fold-change or log2(fold-change) of
the gene. If a <code>log2FoldChange</code> column is provided, the
<code>foldChange</code> column will be ignored. If only a
<code>foldChange</code> column is provided, a
<code>log2FoldChange</code> will be constructed as
<code>log2(foldChange)</code>.</p></li>
</ul>
<p>Any other column present in the data-frame will be ignored. An
example of valid input is shown below:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 7</span></span></span>
<span><span class="co">#&gt;   gene                 chromosome   padj log2FoldChange   start     end label   </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGSC0003DMG400013995 ST4.03ch00  0.972       0.000<span style="text-decoration: underline;">728</span>  <span style="text-decoration: underline;">174</span>833  <span style="text-decoration: underline;">179</span>264 Serine …</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PGSC0003DMG400045238 ST4.03ch00  0.922      -<span style="color: #BB0000;">0.000</span><span style="color: #BB0000; text-decoration: underline;">878</span>  <span style="text-decoration: underline;">461</span>366  <span style="text-decoration: underline;">461</span>756 N-acyle…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PGSC0003DMG400045392 ST4.03ch00 <span style="color: #BB0000;">NA</span>           0.002<span style="text-decoration: underline;">87</span>   <span style="text-decoration: underline;">690</span>521  <span style="text-decoration: underline;">692</span>816 Gene of…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PGSC0003DMG400022052 ST4.03ch00  0.609       0.029<span style="text-decoration: underline;">3</span>    <span style="text-decoration: underline;">804</span>763  <span style="text-decoration: underline;">805</span>569 Conserv…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PGSC0003DMG400022093 ST4.03ch00  0.176       0.147    1<span style="text-decoration: underline;">247</span>383 1<span style="text-decoration: underline;">252</span>635 Conserv…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PGSC0003DMG402018893 ST4.03ch00  0.950       0.001<span style="text-decoration: underline;">25</span>  1<span style="text-decoration: underline;">510</span>542 1<span style="text-decoration: underline;">513</span>352 Stricto…</span></span></code></pre></div>
<p>(Note that in the example dataset, some genes have missing values in
the <code>padj</code> column; this corresponds to genes that have been
filtered out via <a href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#pvaluesNA" class="external-link">independent
filtering in the DEseq2 package</a>).</p>
</div>
<div class="section level3">
<h3 id="candidate-genes">Candidate genes<a class="anchor" aria-label="anchor" href="#candidate-genes"></a>
</h3>
<p>A list of candidate genes (e.g. genes previously found associated
with a trait of interest based on literature search) can be provided as
a tibble or data-frame, with one row per gene. This data-frame can also
contain variants of interest (see below). The data-frame should contain
at least the following columns:</p>
<ul>
<li><p><code>chromosome</code>: character column, giving the ID of the
chromosome on which each gene is located;</p></li>
<li><p><code>start</code> and <code>end</code>: numeric columns, giving
the starting and end position of the gene along the chromosome (in bp).
These two columns will be used to calculate the position of the gene as
the half-way point between the start and end of the gene. For genomic
variants or markers, simply set both the <code>start</code> and
<code>end</code> columns to the physical position of the
marker.</p></li>
<li><p><code>name</code>: character column, giving the name of the
candidate gene that will be displayed in the HIDECAN plot. Set to
<code>NA</code> for any subset of genes to remove their label in the
plot (can help if many genes are very close, to avoid cluttering the
plot).</p></li>
</ul>
<p>Any other column present in the data-frame will be ignored. An
example of valid input is shown below:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 6</span></span></span>
<span><span class="co">#&gt;   id                   chromosome    start      end name       gene_name        </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGSC0003DMG400003155 ST4.03ch03 46<span style="text-decoration: underline;">757</span>152 46<span style="text-decoration: underline;">762</span>127 4CL        4-coumarate-CoA …</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PGSC0003DMG400014223 ST4.03ch03 57<span style="text-decoration: underline;">466</span>692 57<span style="text-decoration: underline;">469</span>946 4CL2       4-coumarate-CoA …</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PGSC0003DMG400011189 ST4.03ch07  1<span style="text-decoration: underline;">001</span>854  1<span style="text-decoration: underline;">006</span>278 HQT        HQT              </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PGSC0003DMG400005492 ST4.03ch05 36<span style="text-decoration: underline;">342</span>746 36<span style="text-decoration: underline;">347</span>409 PAL        phenylalanine am…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PGSC0003DMG400005279 ST4.03ch05 42<span style="text-decoration: underline;">523</span>943 42<span style="text-decoration: underline;">525</span>912 peroxidase peroxidase       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PGSC0003DMG400007782 ST4.03ch03 38<span style="text-decoration: underline;">537</span>202 38<span style="text-decoration: underline;">540</span>209 PHO1A      PHO1A</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="creating-a-hidecan-plot">Creating a HIDECAN plot<a class="anchor" aria-label="anchor" href="#creating-a-hidecan-plot"></a>
</h2>
<p>The <code><a href="../reference/hidecan_plot.html">hidecan_plot()</a></code> function creates a HIDECAN plot. It
takes as input the data-frames presented above, as well as the score and
log2(fold-change) thresholds used to select the significant markers and
genes.</p>
<p>In this example, we only show markers with a score above 4, which
corresponds to a p-value below <span class="math inline">\(1\times10^{-4}\)</span>, and genes with a score
above 1.3, which corresponds to a p-value of 0.05. We don’t place any
threshold on the log2(fold-change) of the genes:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,          <span class="co">## data-frame of GWAS results          </span></span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,              <span class="co">## data-frame of DE results              </span></span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,            <span class="co">## data-frame of candidate genes            </span></span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,  <span class="co">## sign. threshold for GWAS</span></span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,      <span class="co">## sign. threshold for DE</span></span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>                    <span class="co">## log2FC threshold for DE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-plot-1.png" width="960"></p>
<p>Note that it is possible to provide only a subset of the possible
input data, e.g. only GWAS results and a list of candidate genes:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,          </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,            </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="fl">4</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-plot-gwas-can-only-1.png" width="960"></p>
</div>
<div class="section level2">
<h2 id="removing-empty-chromosomes">Removing empty chromosomes<a class="anchor" aria-label="anchor" href="#removing-empty-chromosomes"></a>
</h2>
<p>By default, the HIDECAN plot shows all chromosomes present in the
input data. However, it is possible that some of the chromosomes appear
empty, as they do not contain any significant gene or marker, nor any
candidate gene. In this case, it is possible to exclude such “empty”
chromosomes from the HIDECAN plot, through the
<code>remove_empty_chrom</code> argument.</p>
<p>We will demonstrate that by increasing the score threshold applied to
the GWAS results, in order to get fewer significant markers. In this
case, chromosomes 0, 6, 9 and 10 do not contain any significant marker
of gene of interest:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Chromosomes 0, 6, 9 and 10 are empty</span></span>
<span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,          </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,            </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="fl">5</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-plot-with-empty-chrom-1.png" width="960"></p>
<p>By setting the <code>remove_empty_chrom</code> argument to
<code>TRUE</code>, these chromosomes will be removed from the plot:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,          </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,            </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  remove_empty_chrom <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-plot-without-empty-chrom-1.png" width="960"></p>
</div>
<div class="section level2">
<h2 id="selecting-chromosomes-and-genomic-positions">Selecting chromosomes and genomic positions<a class="anchor" aria-label="anchor" href="#selecting-chromosomes-and-genomic-positions"></a>
</h2>
<p>It is possible to specify which chromosomes should be represented in
the HIDECAN plot, via the <code>chroms</code> argument. For example,
with the following command we restrict the plot to chromosomes 7 and
8:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,                    </span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,                          </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,                  </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,  </span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,      </span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  chroms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ST4.03ch07"</span>, <span class="st">"ST4.03ch08"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-select-chroms-1.png" width="960"></p>
<p>We can also “zoom in” on some or all chromosomes, through the
<code>chrom_limits</code> argument. To zoom in on all chromosomes at
once, we pass to the <code>chrom_limits</code> argument an integer
vector of length 2, which gives the lower and upper limits in bp to use.
For example here, we focus on the 10-20Mb region of each chromosome:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,                    </span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,                          </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,                  </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,  </span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,      </span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  chrom_limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10e6</span>, <span class="fl">20e6</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-chrom-limits-all-chrom-1.png" width="960"></p>
<p>Alternatively, we can apply different limits to some of the
chromosomes, by passing a named list to the argument. The names of the
list should match the chromosomes name, and each element should be an
integer vector of length 2 giving the lower and upper limits in bp to
use for the corresponding chromosome. For example, we will focus on the
10-20Mb region for chromosome 1, and the 30-40Mb region for chromosome
5, and leave all other chromosomes as is:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,                    </span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,                          </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,                  </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,  </span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,      </span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  chrom_limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"ST4.03ch01"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10e6</span>, <span class="fl">20e6</span><span class="op">)</span>,</span>
<span>                      <span class="st">"ST4.03ch05"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">30e6</span>, <span class="fl">40e6</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-chrom-limits-some-chrom-1.png" width="960"></p>
<p>The two options <code>chroms</code> and <code>chrom_limits</code> can
be used together:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,                    </span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,                          </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,                  </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,  </span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,      </span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  chroms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ST4.03ch07"</span>, <span class="st">"ST4.03ch08"</span><span class="op">)</span>,</span>
<span>  chrom_limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"ST4.03ch07"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">50e6</span>, <span class="fl">55e6</span><span class="op">)</span>,</span>
<span>                      <span class="st">"ST4.03ch08"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">45e6</span>, <span class="fl">50e6</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-select-chroms-and-chrom-lims-1.png" width="960"></p>
</div>
<div class="section level2">
<h2 id="colour-genes-by-log2fold-change">Colour genes by log2(fold-change)<a class="anchor" aria-label="anchor" href="#colour-genes-by-log2fold-change"></a>
</h2>
<p>By default, in a HIDECAN plot, the points representing both
significant markers and DE genes are coloured according to their GWAS/DE
score. However, it is possible to colour the DE genes by their
log2(fold-change) value instead, by setting the
<code>colour_genes_by_score</code> argument to <code>FALSE</code>:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,          </span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,              </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,            </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,</span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  colour_genes_by_score <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-genes-colour-log2fc-1.png" width="960"></p>
<p>Genes with a negative log2(fold-change) will be represented with a
shade of blue, and genes with a positive log2(fold-change) will be
represented with a shade of red.</p>
</div>
<div class="section level2">
<h2 id="more-than-one-gwas-de-or-candidate-gene-list">More than one GWAS, DE or candidate gene list<a class="anchor" aria-label="anchor" href="#more-than-one-gwas-de-or-candidate-gene-list"></a>
</h2>
<p>The <code><a href="../reference/hidecan_plot.html">hidecan_plot()</a></code> function can take as an input lists
of data-frames for GWAS results, DE results or candidate genes. This
way, it is possible to visualise more than one GWAS or DE analyses at
once, for example if investigating several traits at once or comparing
more than two treatment groups.</p>
<p>For this example, we’ll focus on chromosomes 7 and 8 (only for
clarity of the plot):</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://purrr.tidyverse.org/" class="external-link">purrr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Retaining only markers and genes on chromosomes 7 and 8</span></span>
<span><span class="va">x_small</span> <span class="op">&lt;-</span> <span class="va">x</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="op">~</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">.x</span>, <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">chromosome</span>, <span class="st">"(07|08)"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>We’ll create a second data-frame of GWAS results by shuffling the
marker scores in the example dataset:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Creating a second GWAS result tibble by shuffling </span></span>
<span><span class="co">## the marker scores from the original data</span></span>
<span><span class="va">gwas_1</span> <span class="op">&lt;-</span> <span class="va">x_small</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">gwas_2</span> <span class="op">&lt;-</span> <span class="va">gwas_1</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">score</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>We can pass both GWAS results data-frames to the
<code><a href="../reference/hidecan_plot.html">hidecan_plot()</a></code> function as a list:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">gwas_1</span>, <span class="va">gwas_2</span><span class="op">)</span>,</span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,</span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-multiple-gwas-input-1.png" width="960"></p>
<p>By default, the two GWAS tracks will be given unique y-axis labels,
as can be seen above. It is possible to customise this by naming the
elements in the input list:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Trait 1"</span> <span class="op">=</span> <span class="va">gwas_1</span>, </span>
<span>                   <span class="st">"Trait 2"</span> <span class="op">=</span> <span class="va">gwas_2</span><span class="op">)</span>,</span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,</span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/multiple-gwas-input-named-1.png" width="960"></p>
</div>
<div class="section level2">
<h2 id="defining-chromosomes-length">Defining chromosomes length<a class="anchor" aria-label="anchor" href="#defining-chromosomes-length"></a>
</h2>
<p>By default, the <code><a href="../reference/hidecan_plot.html">hidecan_plot()</a></code> function calculates the
length of the different chromosomes based on the input data, by looking
at the maximum position of genes and markers on each chromosome.
However, it is also possible to pass on a tibble of chromosome length
(in bp) through the <code>chrom_length</code> argument.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Chromosomes length as recorded in Ensembl Plants</span></span>
<span><span class="va">potato_chrom_length</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  ST4.03ch00 <span class="op">=</span> <span class="fl">45813526</span>,</span>
<span>  ST4.03ch01 <span class="op">=</span> <span class="fl">88663952</span>,</span>
<span>  ST4.03ch02 <span class="op">=</span> <span class="fl">48614681</span>,</span>
<span>  ST4.03ch03 <span class="op">=</span> <span class="fl">62190286</span>,</span>
<span>  ST4.03ch04 <span class="op">=</span> <span class="fl">72208621</span>,</span>
<span>  ST4.03ch05 <span class="op">=</span> <span class="fl">52070158</span>,</span>
<span>  ST4.03ch06 <span class="op">=</span> <span class="fl">59532096</span>,</span>
<span>  ST4.03ch07 <span class="op">=</span> <span class="fl">56760843</span>,</span>
<span>  ST4.03ch08 <span class="op">=</span> <span class="fl">56938457</span>,</span>
<span>  ST4.03ch09 <span class="op">=</span> <span class="fl">61540751</span>,</span>
<span>  ST4.03ch10 <span class="op">=</span> <span class="fl">59756223</span>,</span>
<span>  ST4.03ch11 <span class="op">=</span> <span class="fl">45475667</span>,</span>
<span>  ST4.03ch12 <span class="op">=</span> <span class="fl">61165649</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="co">## turn a named vector into a tibble</span></span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/enframe.html" class="external-link">enframe</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"chromosome"</span>,</span>
<span>          value <span class="op">=</span> <span class="st">"length"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">potato_chrom_length</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 2</span></span></span>
<span><span class="co">#&gt;   chromosome   length</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> ST4.03ch00 45<span style="text-decoration: underline;">813</span>526</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> ST4.03ch01 88<span style="text-decoration: underline;">663</span>952</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> ST4.03ch02 48<span style="text-decoration: underline;">614</span>681</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> ST4.03ch03 62<span style="text-decoration: underline;">190</span>286</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> ST4.03ch04 72<span style="text-decoration: underline;">208</span>621</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> ST4.03ch05 52<span style="text-decoration: underline;">070</span>158</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,          </span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,              </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,</span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  chrom_length <span class="op">=</span> <span class="va">potato_chrom_length</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-chrom-length-1.png" width="960"></p>
<p>Note that in this case we can’t really see the difference with the
computed chromosome length values.</p>
</div>
<div class="section level2">
<h2 id="controlling-the-plot-properties">Controlling the plot properties<a class="anchor" aria-label="anchor" href="#controlling-the-plot-properties"></a>
</h2>
<p>The <code><a href="../reference/hidecan_plot.html">hidecan_plot()</a></code> function offers several arguments to
control different aspects of the HIDECAN plot. For example, it is
possible to specify the number of rows or columns the plot should have,
through the <code>n_rows</code> and <code>n_cols</code> arguments. Note
that only one of these arguments will be considered (<code>n_rows</code>
takes precedence):</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Specifying the number of rows</span></span>
<span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,          </span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,              </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,            </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,</span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  n_rows <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-nrows-1.png" width="960"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Specifying the number of columns</span></span>
<span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,          </span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,              </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,            </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.005</span><span class="op">)</span>,</span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  n_cols <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-ncols-1.png" width="960"></p>
<p>In addition, it is possible to:</p>
<ul>
<li><p>add a title and subtitle to the plot (<code>title</code> and
<code>subtitle</code> arguments);</p></li>
<li><p>control the position of the legend (<code>legend_position</code>
argument);</p></li>
<li><p>control the size of the points (<code>point_size</code>
argument);</p></li>
<li><p>control the size of the labels and padding around the text for
the candidate genes labels (<code>label_size</code> and
<code>label_padding</code> arguments).</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="viewport-error">Viewport error<a class="anchor" aria-label="anchor" href="#viewport-error"></a>
</h2>
<p>If you are working on RStudio, you may encounter the following
error:</p>
<pre><code><span><span class="co">#&gt; Error in grid.Call(C_convert, x, as.integer(whatfrom), as.integer(whatto), : Viewport has zero dimension(s)</span></span></code></pre>
<p>This is caused by the plotting window being too small. Try increasing
the size of the plotting window in the RStudio console. Alternatively,
you can save the plot into an R object, then use
<code><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggplot2::ggsave()</a></code> to save it into a file:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,</span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave</a></span><span class="op">(</span><span class="st">"hidecan_plot.pdf"</span>, <span class="va">p</span>, width <span class="op">=</span> <span class="fl">10</span>, height <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Olivia Angelin-Bonnet.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
