<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="hidecan">
<title>Creating a HIDECAN plot step by step • hidecan</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Creating a HIDECAN plot step by step">
<meta property="og:description" content="hidecan">
<meta property="og:image" content="https://plantandfoodresearch.github.io/hidecan/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">hidecan</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/hidecan.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/hidecan-step-by-step.html">Step-by-step</a>
    <a class="dropdown-item" href="../articles/web_only/gwaspoly_output.html">GWASpoly output</a>
    <a class="dropdown-item" href="../articles/web_only/shiny-app.html">Shiny app</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/PlantandFoodResearch/hidecan/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Creating a HIDECAN plot step by step</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/PlantandFoodResearch/hidecan/blob/HEAD/vignettes/hidecan-step-by-step.Rmd" class="external-link"><code>vignettes/hidecan-step-by-step.Rmd</code></a></small>
      <div class="d-none name"><code>hidecan-step-by-step.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://plantandfoodresearch.github.io/hidecan/">hidecan</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://purrr.tidyverse.org/" class="external-link">purrr</a></span><span class="op">)</span></span></code></pre></div>
<p>The <code><a href="../reference/hidecan_plot.html">hidecan_plot()</a></code> function is really a wrapper that
performs a series of steps to go from GWAS/DE results and candidate
genes data-frames to a HIDECAN (gg)plot. In this vignette, we present
the different steps performed by the wrapper function. This can be
useful for users who want more control over the process. We’ll work with
the example dataset provided with the package</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_example_data.html">get_example_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 3</span></span>
<span><span class="co">#&gt;  $ GWAS: tibble [35,481 × 4] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;   ..$ id        : chr [1:35481] "ST4.03ch00_45467783" "ST4.03ch01_88589716" "ST4.03ch02_48614228" "ST4.03ch03_62263578" ...</span></span>
<span><span class="co">#&gt;   ..$ chromosome: chr [1:35481] "ST4.03ch00" "ST4.03ch01" "ST4.03ch02" "ST4.03ch03" ...</span></span>
<span><span class="co">#&gt;   ..$ position  : num [1:35481] 45467783 88589716 48614228 62263578 72139135 ...</span></span>
<span><span class="co">#&gt;   ..$ score     : num [1:35481] 0.191 1.835 0.381 0.661 0.64 ...</span></span>
<span><span class="co">#&gt;  $ DE  : tibble [10,671 × 7] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;   ..$ gene          : chr [1:10671] "PGSC0003DMG400032056" "PGSC0003DMG400018039" "PGSC0003DMG400020231" "PGSC0003DMG400009197" ...</span></span>
<span><span class="co">#&gt;   ..$ chromosome    : chr [1:10671] "ST4.03ch00" "ST4.03ch01" "ST4.03ch02" "ST4.03ch03" ...</span></span>
<span><span class="co">#&gt;   ..$ padj          : num [1:10671] 0.787 0.63 0.864 0.53 0.975 ...</span></span>
<span><span class="co">#&gt;   ..$ log2FoldChange: num [1:10671] 0.01143 0.00529 0.00362 0.03197 0.00225 ...</span></span>
<span><span class="co">#&gt;   ..$ start         : num [1:10671] 45813195 88623473 48563271 62256322 72168842 ...</span></span>
<span><span class="co">#&gt;   ..$ end           : num [1:10671] 45813526 88627702 48578978 62258929 72170119 ...</span></span>
<span><span class="co">#&gt;   ..$ label         : chr [1:10671] "Protein transporter" "PhD-finger protein" "Acetyl-CoA synthetase" "Phosphatidylcholine transfer protein" ...</span></span>
<span><span class="co">#&gt;  $ CAN : tibble [32 × 6] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;   ..$ id        : chr [1:32] "PGSC0003DMG400003155" "PGSC0003DMG400014223" "PGSC0003DMG400011189" "PGSC0003DMG400005492" ...</span></span>
<span><span class="co">#&gt;   ..$ chromosome: chr [1:32] "ST4.03ch03" "ST4.03ch03" "ST4.03ch07" "ST4.03ch05" ...</span></span>
<span><span class="co">#&gt;   ..$ start     : num [1:32] 46757152 57466692 1001854 36342746 42523943 ...</span></span>
<span><span class="co">#&gt;   ..$ end       : num [1:32] 46762127 57469946 1006278 36347409 42525912 ...</span></span>
<span><span class="co">#&gt;   ..$ name      : chr [1:32] "4CL" "4CL2" "HQT" "PAL" ...</span></span>
<span><span class="co">#&gt;   ..$ gene_name : chr [1:32] "4-coumarate-CoA ligase" "4-coumarate-CoA ligase 2" "HQT" "phenylalanine ammonia-lyase" ...</span></span></code></pre></div>
<div class="section level2">
<h2 id="formatting-input-data">Formatting input data<a class="anchor" aria-label="anchor" href="#formatting-input-data"></a>
</h2>
<p>Under the hood, the <code>hidecan</code> package relies on S3
classes, which are really just tibbles with specific columns. The
constructors for these S3 classes perform a series of checks and
computations to make sure that all of the required columns (such as
chromosome, score, position) are present in the data.</p>
<ul>
<li>GWAS results data-frames are turned into <code>GWAS_data</code>
objects through the <code><a href="../reference/GWAS_data.html">GWAS_data()</a></code> constructor:</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwas_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWAS_data.html">GWAS_data</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">gwas_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "GWAS_data"  "tbl_df"     "tbl"        "data.frame"</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gwas_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;   id                  chromosome position score</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> ST4.03ch00_45467783 ST4.03ch00 45<span style="text-decoration: underline;">467</span>783 0.191</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> ST4.03ch01_88589716 ST4.03ch01 88<span style="text-decoration: underline;">589</span>716 1.84 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> ST4.03ch02_48614228 ST4.03ch02 48<span style="text-decoration: underline;">614</span>228 0.381</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> ST4.03ch03_62263578 ST4.03ch03 62<span style="text-decoration: underline;">263</span>578 0.661</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> ST4.03ch04_72139135 ST4.03ch04 72<span style="text-decoration: underline;">139</span>135 0.640</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> ST4.03ch05_52040302 ST4.03ch05 52<span style="text-decoration: underline;">040</span>302 0.346</span></span></code></pre></div>
<ul>
<li>DE results data-frames are turned into <code>DE_data</code> objects
through the <code><a href="../reference/DE_data.html">DE_data()</a></code> constructor:</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">de_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DE_data.html">DE_data</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">de_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "DE_data"    "tbl_df"     "tbl"        "data.frame"</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">de_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 9</span></span></span>
<span><span class="co">#&gt;   gene     chromosome   padj log2FoldChange  start    end label   score position</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGSC000… ST4.03ch00  0.787       0.011<span style="text-decoration: underline;">4</span>   4.58<span style="color: #949494;">e</span>7 4.58<span style="color: #949494;">e</span>7 Prot…  0.104    4.58<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PGSC000… ST4.03ch01  0.630       0.005<span style="text-decoration: underline;">29</span>  8.86<span style="color: #949494;">e</span>7 8.86<span style="color: #949494;">e</span>7 PhD-…  0.201    8.86<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PGSC000… ST4.03ch02  0.864       0.003<span style="text-decoration: underline;">62</span>  4.86<span style="color: #949494;">e</span>7 4.86<span style="color: #949494;">e</span>7 Acet…  0.063<span style="text-decoration: underline;">7</span>   4.86<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PGSC000… ST4.03ch03  0.530       0.032<span style="text-decoration: underline;">0</span>   6.23<span style="color: #949494;">e</span>7 6.23<span style="color: #949494;">e</span>7 Phos…  0.276    6.23<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PGSC000… ST4.03ch04  0.975       0.002<span style="text-decoration: underline;">25</span>  7.22<span style="color: #949494;">e</span>7 7.22<span style="color: #949494;">e</span>7 Cons…  0.010<span style="text-decoration: underline;">9</span>   7.22<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PGSC000… ST4.03ch05 <span style="color: #BB0000;">NA</span>          -<span style="color: #BB0000;">0.000</span><span style="color: #BB0000; text-decoration: underline;">726</span> 5.20<span style="color: #949494;">e</span>7 5.20<span style="color: #949494;">e</span>7 Cons… <span style="color: #BB0000;">NA</span>        5.20<span style="color: #949494;">e</span>7</span></span></code></pre></div>
<ul>
<li>Candidate genes data-frames are turned into <code>CAN_data</code>
objects through the <code><a href="../reference/CAN_data.html">CAN_data()</a></code> constructor:</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## CAN_data constructor</span></span>
<span><span class="va">can_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CAN_data.html">CAN_data</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">can_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "CAN_data"   "tbl_df"     "tbl"        "data.frame"</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">can_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 7</span></span></span>
<span><span class="co">#&gt;   id                   chromosome    start      end name      gene_name position</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGSC0003DMG400003155 ST4.03ch03 46<span style="text-decoration: underline;">757</span>152 46<span style="text-decoration: underline;">762</span>127 4CL       4-coumar…   4.68<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PGSC0003DMG400014223 ST4.03ch03 57<span style="text-decoration: underline;">466</span>692 57<span style="text-decoration: underline;">469</span>946 4CL2      4-coumar…   5.75<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PGSC0003DMG400011189 ST4.03ch07  1<span style="text-decoration: underline;">001</span>854  1<span style="text-decoration: underline;">006</span>278 HQT       HQT         1.00<span style="color: #949494;">e</span>6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PGSC0003DMG400005492 ST4.03ch05 36<span style="text-decoration: underline;">342</span>746 36<span style="text-decoration: underline;">347</span>409 PAL       phenylal…   3.63<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PGSC0003DMG400005279 ST4.03ch05 42<span style="text-decoration: underline;">523</span>943 42<span style="text-decoration: underline;">525</span>912 peroxida… peroxida…   4.25<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PGSC0003DMG400007782 ST4.03ch03 38<span style="text-decoration: underline;">537</span>202 38<span style="text-decoration: underline;">540</span>209 PHO1A     PHO1A       3.85<span style="color: #949494;">e</span>7</span></span></code></pre></div>
<p>These constructors will throw an error if a required column is
missing from the input data (e.g. no chromosome column):</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwas_wrong_input</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">chromosome</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/GWAS_data.html">GWAS_data</a></span><span class="op">(</span><span class="va">gwas_wrong_input</span><span class="op">)</span></span>
<span><span class="co">#&gt; Error: Input data-frame is missing the following columns: 'chromosome'.</span></span></code></pre></div>
<p>They will also compute marker or gene scores from adjusted p-values
if necessary (see the <a href="hidecan.html#input-data">Input data</a>
section of the hidecan vignette). For example, for DE results, if we
provide a <code>padj</code> column (with the adjusted p-values of the
genes) rather than a <code>score</code> column, the constructor will
compute the <code>score</code> column based on the <code>padj</code>
column. You can also notice that a <code>position</code> column is
computed based on the start and end of the genes:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Input tibble</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 7</span></span></span>
<span><span class="co">#&gt;   gene                 chromosome   padj log2FoldChange    start      end label </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGSC0003DMG400032056 ST4.03ch00  0.787       0.011<span style="text-decoration: underline;">4</span>   45<span style="text-decoration: underline;">813</span>195 45<span style="text-decoration: underline;">813</span>526 Prote…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PGSC0003DMG400018039 ST4.03ch01  0.630       0.005<span style="text-decoration: underline;">29</span>  88<span style="text-decoration: underline;">623</span>473 88<span style="text-decoration: underline;">627</span>702 PhD-f…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PGSC0003DMG400020231 ST4.03ch02  0.864       0.003<span style="text-decoration: underline;">62</span>  48<span style="text-decoration: underline;">563</span>271 48<span style="text-decoration: underline;">578</span>978 Acety…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PGSC0003DMG400009197 ST4.03ch03  0.530       0.032<span style="text-decoration: underline;">0</span>   62<span style="text-decoration: underline;">256</span>322 62<span style="text-decoration: underline;">258</span>929 Phosp…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PGSC0003DMG403025662 ST4.03ch04  0.975       0.002<span style="text-decoration: underline;">25</span>  72<span style="text-decoration: underline;">168</span>842 72<span style="text-decoration: underline;">170</span>119 Conse…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PGSC0003DMG400023316 ST4.03ch05 <span style="color: #BB0000;">NA</span>          -<span style="color: #BB0000;">0.000</span><span style="color: #BB0000; text-decoration: underline;">726</span> 52<span style="text-decoration: underline;">039</span>916 52<span style="text-decoration: underline;">040</span>326 Conse…</span></span>
<span></span>
<span><span class="co">## Output of the DE_data constructor</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">de_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 9</span></span></span>
<span><span class="co">#&gt;   gene     chromosome   padj log2FoldChange  start    end label   score position</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGSC000… ST4.03ch00  0.787       0.011<span style="text-decoration: underline;">4</span>   4.58<span style="color: #949494;">e</span>7 4.58<span style="color: #949494;">e</span>7 Prot…  0.104    4.58<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PGSC000… ST4.03ch01  0.630       0.005<span style="text-decoration: underline;">29</span>  8.86<span style="color: #949494;">e</span>7 8.86<span style="color: #949494;">e</span>7 PhD-…  0.201    8.86<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PGSC000… ST4.03ch02  0.864       0.003<span style="text-decoration: underline;">62</span>  4.86<span style="color: #949494;">e</span>7 4.86<span style="color: #949494;">e</span>7 Acet…  0.063<span style="text-decoration: underline;">7</span>   4.86<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PGSC000… ST4.03ch03  0.530       0.032<span style="text-decoration: underline;">0</span>   6.23<span style="color: #949494;">e</span>7 6.23<span style="color: #949494;">e</span>7 Phos…  0.276    6.23<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PGSC000… ST4.03ch04  0.975       0.002<span style="text-decoration: underline;">25</span>  7.22<span style="color: #949494;">e</span>7 7.22<span style="color: #949494;">e</span>7 Cons…  0.010<span style="text-decoration: underline;">9</span>   7.22<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PGSC000… ST4.03ch05 <span style="color: #BB0000;">NA</span>          -<span style="color: #BB0000;">0.000</span><span style="color: #BB0000; text-decoration: underline;">726</span> 5.20<span style="color: #949494;">e</span>7 5.20<span style="color: #949494;">e</span>7 Cons… <span style="color: #BB0000;">NA</span>        5.20<span style="color: #949494;">e</span>7</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="computing-chromosome-length">Computing chromosome length<a class="anchor" aria-label="anchor" href="#computing-chromosome-length"></a>
</h2>
<p>Once the input datasets have been formatted appropriately, they are
used to compute the length of the chromosomes present in the data. This
is done through the <code><a href="../reference/combine_chrom_length.html">combine_chrom_length()</a></code> function, which
is applied to a list of <code>GWAS_data</code>, <code>DE_data</code> or
<code>CAN_data</code> objects:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chrom_length</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/combine_chrom_length.html">combine_chrom_length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">gwas_data</span>,</span>
<span>                                          <span class="va">de_data</span>,</span>
<span>                                          <span class="va">can_data</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">chrom_length</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 13 × 2</span></span></span>
<span><span class="co">#&gt;    chromosome   length</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> ST4.03ch00 45<span style="text-decoration: underline;">813</span>526</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> ST4.03ch01 88<span style="text-decoration: underline;">627</span>702</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> ST4.03ch02 48<span style="text-decoration: underline;">614</span>228</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> ST4.03ch03 62<span style="text-decoration: underline;">263</span>578</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> ST4.03ch04 72<span style="text-decoration: underline;">170</span>119</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> ST4.03ch05 52<span style="text-decoration: underline;">040</span>326</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> ST4.03ch06 59<span style="text-decoration: underline;">476</span>545</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> ST4.03ch07 56<span style="text-decoration: underline;">715</span>111</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> ST4.03ch08 56<span style="text-decoration: underline;">937</span>627</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> ST4.03ch09 61<span style="text-decoration: underline;">539</span>681</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> ST4.03ch10 59<span style="text-decoration: underline;">687</span>482</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> ST4.03ch11 45<span style="text-decoration: underline;">409</span>456</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> ST4.03ch12 61<span style="text-decoration: underline;">152</span>223</span></span></code></pre></div>
<p>The function works by calling for each element in the list the
<code><a href="../reference/compute_chrom_length.html">compute_chrom_length()</a></code> function. The function, according to
whether the input is a tibble of markers (<code>GWAS_data</code>) or
genes (<code>DE_data</code> or <code>CAN_data</code>), looks for the
maximum value in either the <code>position</code> column (for markers)
or the <code>end</code> column (for genes).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/compute_chrom_length.html">compute_chrom_length</a></span><span class="op">(</span><span class="va">gwas_data</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 2</span></span></span>
<span><span class="co">#&gt;   chromosome   length</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> ST4.03ch00 45<span style="text-decoration: underline;">467</span>783</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> ST4.03ch01 88<span style="text-decoration: underline;">589</span>716</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> ST4.03ch02 48<span style="text-decoration: underline;">614</span>228</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/compute_chrom_length.html">compute_chrom_length</a></span><span class="op">(</span><span class="va">de_data</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 2</span></span></span>
<span><span class="co">#&gt;   chromosome   length</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> ST4.03ch00 45<span style="text-decoration: underline;">813</span>526</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> ST4.03ch01 88<span style="text-decoration: underline;">627</span>702</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> ST4.03ch02 48<span style="text-decoration: underline;">578</span>978</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="applying-threshold">Applying threshold<a class="anchor" aria-label="anchor" href="#applying-threshold"></a>
</h2>
<p>Next, the GWAS and DE results tibbles are filtered according to a
threshold, in order to retain the significant markers or genes. This is
done through the <code><a href="../reference/apply_threshold.html">apply_threshold()</a></code> function. This function
has two (rather self-explanatory) arguments: <code>score_thr</code> and
<code>log2fc_thr</code>.</p>
<p>When applied to a <code>GWAS_data</code> object, the function filters
markers with a score above the value set with <code>score_thr</code>
argument (the <code>log2fc_thr</code> argument is ignored), and returns
an object of class <code>GWAS_data_thr</code>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">gwas_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 35481     4</span></span>
<span></span>
<span><span class="va">gwas_data_thr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/apply_threshold.html">apply_threshold</a></span><span class="op">(</span><span class="va">gwas_data</span>, </span>
<span>                                 score_thr <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">gwas_data_thr</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "GWAS_data_thr" "tbl_df"        "tbl"           "data.frame"</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">gwas_data_thr</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 37  4</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gwas_data_thr</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;   id                  chromosome position score</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> ST4.03ch00_22680252 ST4.03ch00 22<span style="text-decoration: underline;">680</span>252  4.41</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> ST4.03ch01_6317643  ST4.03ch01  6<span style="text-decoration: underline;">317</span>643  4.15</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> ST4.03ch01_7671100  ST4.03ch01  7<span style="text-decoration: underline;">671</span>100  4.43</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> ST4.03ch01_8653747  ST4.03ch01  8<span style="text-decoration: underline;">653</span>747  4.69</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> ST4.03ch01_12842648 ST4.03ch01 12<span style="text-decoration: underline;">842</span>648  6.85</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> ST4.03ch01_13334335 ST4.03ch01 13<span style="text-decoration: underline;">334</span>335  5.24</span></span></code></pre></div>
<p>For a <code>DE_data</code> object, the apply_threshold function
filters genes based on both their score and log2(fold-change), and
returns an object of class <code>DE_data_thr</code>:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">de_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 10671     9</span></span>
<span></span>
<span><span class="va">de_data_thr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/apply_threshold.html">apply_threshold</a></span><span class="op">(</span><span class="va">de_data</span>, </span>
<span>                               score_thr <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                               log2fc_thr <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">de_data_thr</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "DE_data_thr" "tbl_df"      "tbl"         "data.frame"</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">de_data_thr</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 2 9</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">de_data_thr</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 2 × 9</span></span></span>
<span><span class="co">#&gt;   gene      chromosome    padj log2FoldChange  start    end label score position</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGSC0003… ST4.03ch10 0.004<span style="text-decoration: underline;">80</span>          0.757 3.30<span style="color: #949494;">e</span>7 3.30<span style="color: #949494;">e</span>7 Cons…  2.32   3.30<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PGSC0003… ST4.03ch10 0.003<span style="text-decoration: underline;">05</span>          0.691 5.82<span style="color: #949494;">e</span>7 5.82<span style="color: #949494;">e</span>7 N-ac…  2.52   5.82<span style="color: #949494;">e</span>7</span></span></code></pre></div>
<p>Finally, if applied to a CAN_data object, the
<code><a href="../reference/apply_threshold.html">apply_threshold()</a></code> function simply returns the input tibble
as an object of class <code>CAN_data_thr</code>:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">can_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 32  7</span></span>
<span></span>
<span><span class="va">can_data_thr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/apply_threshold.html">apply_threshold</a></span><span class="op">(</span><span class="va">can_data</span>, </span>
<span>                                score_thr <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                log2fc_thr <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">can_data_thr</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "CAN_data_thr" "tbl_df"       "tbl"          "data.frame"</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">can_data_thr</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 32  7</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">can_data_thr</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 7</span></span></span>
<span><span class="co">#&gt;   id                   chromosome    start      end name      gene_name position</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGSC0003DMG400003155 ST4.03ch03 46<span style="text-decoration: underline;">757</span>152 46<span style="text-decoration: underline;">762</span>127 4CL       4-coumar…   4.68<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PGSC0003DMG400014223 ST4.03ch03 57<span style="text-decoration: underline;">466</span>692 57<span style="text-decoration: underline;">469</span>946 4CL2      4-coumar…   5.75<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PGSC0003DMG400011189 ST4.03ch07  1<span style="text-decoration: underline;">001</span>854  1<span style="text-decoration: underline;">006</span>278 HQT       HQT         1.00<span style="color: #949494;">e</span>6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PGSC0003DMG400005492 ST4.03ch05 36<span style="text-decoration: underline;">342</span>746 36<span style="text-decoration: underline;">347</span>409 PAL       phenylal…   3.63<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PGSC0003DMG400005279 ST4.03ch05 42<span style="text-decoration: underline;">523</span>943 42<span style="text-decoration: underline;">525</span>912 peroxida… peroxida…   4.25<span style="color: #949494;">e</span>7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PGSC0003DMG400007782 ST4.03ch03 38<span style="text-decoration: underline;">537</span>202 38<span style="text-decoration: underline;">540</span>209 PHO1A     PHO1A       3.85<span style="color: #949494;">e</span>7</span></span></code></pre></div>
<p>As for the <code>GWAS_data</code>, <code>DE_data</code> or
<code>CAN_data</code> objects, the <code>GWAS_data_thr</code>,
<code>DE_data_thr</code> and <code>CAN_data_thr</code> objects are
really just tibbles.</p>
</div>
<div class="section level2">
<h2 id="creating-the-hidecan-plot">Creating the HIDECAN plot<a class="anchor" aria-label="anchor" href="#creating-the-hidecan-plot"></a>
</h2>
<p>Finally, the filtered datasets are combined into a list and passed on
to the <code><a href="../reference/create_hidecan_plot.html">create_hidecan_plot()</a></code> function, along with the
tibble of chromosome length. This is the function that generates the
HIDECAN <code>ggplot</code>:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/create_hidecan_plot.html">create_hidecan_plot</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">gwas_data_thr</span>,</span>
<span>       <span class="va">de_data_thr</span>,</span>
<span>       <span class="va">can_data_thr</span><span class="op">)</span>,</span>
<span>  <span class="va">chrom_length</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan-step-by-step_files/figure-html/create-hidecan-plot-1.png" width="960"></p>
<p>This function shares most of its arguments with the
<code><a href="../reference/hidecan_plot.html">hidecan_plot()</a></code> wrapper for controlling different aspects of
the plot.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Olivia Angelin-Bonnet.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
