<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-GB">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>hidecan • hidecan</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="hidecan">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hidecan</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/hidecan.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/hidecan-step-by-step.html">Step-by-step</a></li>
    <li><a class="dropdown-item" href="../articles/web_only/gwaspoly_output.html">GWASpoly output</a></li>
    <li><a class="dropdown-item" href="../articles/web_only/custom_tracks.html">Adding custom tracks</a></li>
    <li><a class="dropdown-item" href="../articles/web_only/shiny-app.html">Shiny app</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/PlantandFoodResearch/hidecan/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>hidecan</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/PlantandFoodResearch/hidecan/blob/main/vignettes/hidecan.Rmd" class="external-link"><code>vignettes/hidecan.Rmd</code></a></small>
      <div class="d-none name"><code>hidecan.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://plantandfoodresearch.github.io/hidecan/">hidecan</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="input-data">Input data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h2>
<p>The <code>hidecan</code> package takes as input tibbles (data-frames)
of QTL mapping, GWAS and DE results or candidate genes. The input
data-frames should contain some mandatory columns, depending on the type
of data. Note that for each type of results, the full list of tested
regions/markers/genes should be passed to <code>hidecan</code>, not only
the significant ones; the package will take care of selecting the
significant results based on user-defined thresholds.</p>
<p>A list of example input datasets can be obtained via the
<code><a href="../reference/get_example_data.html">get_example_data()</a></code> function:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_example_data.html">get_example_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">x</span>, max.level <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 4</span></span>
<span><span class="co">#&gt;  $ QTL : tibble [7 × 6] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ GWAS: tibble [35,481 × 4] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ DE  : tibble [10,671 × 7] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ CAN : tibble [32 × 6] (S3: tbl_df/tbl/data.frame)</span></span></code></pre></div>
<div class="section level3">
<h3 id="qtl-mapping-results">QTL mapping results<a class="anchor" aria-label="anchor" href="#qtl-mapping-results"></a>
</h3>
<p>QTL mapping results should be provided as a tibble or data-frame,
with one row per QTL region. The data-frame should contain at least the
following columns:</p>
<ul>
<li><p><code>chromosome</code>: character column, giving the ID of the
chromosome on which each region is located;</p></li>
<li><p><code>start</code> and <code>end</code>: numeric columns, giving
the starting and end position of the region along the chromosome (in
bp);</p></li>
<li><p>either <code>score</code> or <code>padj</code>: numeric column,
providing either the score (i.e. <code>-log10(p-value)</code>) or the
adjusted p-value of the region If a <code>score</code> column is
provided, the <code>padj</code> column will be ignored. If only a
<code>padj</code> column is provided, a <code>score</code> column will
be constructed as <code>-log10(padj)</code>. Note that this column could
be used to represent the estimated effect of the region, for
example.</p></li>
</ul>
<p>Any other column present in the data-frame will be ignored. An
example of valid input is shown below:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"QTL"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 6</span></span></span>
<span><span class="co">#&gt;   id    chromosome    start      end score name </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> qtl_1 ST4.03ch01  5<span style="text-decoration: underline;">600</span>000 11<span style="text-decoration: underline;">300</span>000   4.1 QTL 1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> qtl_2 ST4.03ch01  9<span style="text-decoration: underline;">500</span>000 14<span style="text-decoration: underline;">000</span>000   5.5 QTL 2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> qtl_3 ST4.03ch04 12<span style="text-decoration: underline;">700</span>000 17<span style="text-decoration: underline;">600</span>000   4.3 QTL 3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> qtl_4 ST4.03ch05 37<span style="text-decoration: underline;">600</span>000 39<span style="text-decoration: underline;">900</span>000   3.2 QTL 4</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> qtl_5 ST4.03ch07 49<span style="text-decoration: underline;">900</span>000 51<span style="text-decoration: underline;">700</span>000   5.1 QTL 5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> qtl_6 ST4.03ch08 45<span style="text-decoration: underline;">000</span>000 47<span style="text-decoration: underline;">000</span>000   5.8 QTL 6</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="gwas-results">GWAS results<a class="anchor" aria-label="anchor" href="#gwas-results"></a>
</h3>
<p>GWAS results should be provided as a tibble or data-frame, with one
row per genetic marker. The data-frame should contain at least the
following columns:</p>
<ul>
<li><p><code>chromosome</code>: character column, giving the ID of the
chromosome on which each marker is located;</p></li>
<li><p><code>position</code>: numeric column, the physical position
along the chromosome (in base pairs - bp) of the marker;</p></li>
<li><p>either <code>score</code> or <code>padj</code>: numeric column,
providing either the score (i.e. <code>-log10(p-value)</code>) or the
adjusted p-value of the region. If a <code>score</code> column is
provided, the <code>padj</code> column will be ignored. If only a
<code>padj</code> column is provided, a <code>score</code> column will
be constructed as <code>-log10(padj)</code>.</p></li>
</ul>
<p>Any other column present in the data-frame will be ignored. An
example of valid input is shown below:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;   id                  chromosome position score</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> ST4.03ch00_45467783 ST4.03ch00 45<span style="text-decoration: underline;">467</span>783 0.191</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> ST4.03ch01_88589716 ST4.03ch01 88<span style="text-decoration: underline;">589</span>716 1.84 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> ST4.03ch02_48614228 ST4.03ch02 48<span style="text-decoration: underline;">614</span>228 0.381</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> ST4.03ch03_62263578 ST4.03ch03 62<span style="text-decoration: underline;">263</span>578 0.661</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> ST4.03ch04_72139135 ST4.03ch04 72<span style="text-decoration: underline;">139</span>135 0.640</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> ST4.03ch05_52040302 ST4.03ch05 52<span style="text-decoration: underline;">040</span>302 0.346</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="differential-expression-results">Differential expression results<a class="anchor" aria-label="anchor" href="#differential-expression-results"></a>
</h3>
<p>DE results should be provided as a tibble or data-frame, with one row
per gene. The data-frame should contain at least the following
columns:</p>
<ul>
<li><p><code>chromosome</code>: character column, giving the ID of the
chromosome on which each gene is located;</p></li>
<li><p><code>start</code> and <code>end</code>: numeric columns, giving
the starting and end position of the gene along the chromosome (in bp).
These two columns will be used to calculate the position of the gene as
the half-way point between the start and end of the gene.</p></li>
<li><p>either <code>score</code> or <code>padj</code>: numeric column,
providing either the score (i.e. <code>-log10(p-value)</code>) or the
adjusted p-value of the gene. If a <code>score</code> column is
provided, the <code>padj</code> column will be ignored. If only a
<code>padj</code> column is provided, a <code>score</code> column will
be constructed as <code>-log10(padj)</code>.</p></li>
<li><p>either <code>foldChange</code> or <code>log2FoldChange</code>:
numeric column, giving either the fold-change or log2(fold-change) of
the gene. If a <code>log2FoldChange</code> column is provided, the
<code>foldChange</code> column will be ignored. If only a
<code>foldChange</code> column is provided, a
<code>log2FoldChange</code> will be constructed as
<code>log2(foldChange)</code>.</p></li>
</ul>
<p>Any other column present in the data-frame will be ignored. An
example of valid input is shown below:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 7</span></span></span>
<span><span class="co">#&gt;   gene                 chromosome   padj log2FoldChange    start      end label </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGSC0003DMG400032056 ST4.03ch00  0.787       0.011<span style="text-decoration: underline;">4</span>   45<span style="text-decoration: underline;">813</span>195 45<span style="text-decoration: underline;">813</span>526 Prote…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PGSC0003DMG400018039 ST4.03ch01  0.630       0.005<span style="text-decoration: underline;">29</span>  88<span style="text-decoration: underline;">623</span>473 88<span style="text-decoration: underline;">627</span>702 PhD-f…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PGSC0003DMG400020231 ST4.03ch02  0.864       0.003<span style="text-decoration: underline;">62</span>  48<span style="text-decoration: underline;">563</span>271 48<span style="text-decoration: underline;">578</span>978 Acety…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PGSC0003DMG400009197 ST4.03ch03  0.530       0.032<span style="text-decoration: underline;">0</span>   62<span style="text-decoration: underline;">256</span>322 62<span style="text-decoration: underline;">258</span>929 Phosp…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PGSC0003DMG403025662 ST4.03ch04  0.975       0.002<span style="text-decoration: underline;">25</span>  72<span style="text-decoration: underline;">168</span>842 72<span style="text-decoration: underline;">170</span>119 Conse…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PGSC0003DMG400023316 ST4.03ch05 <span style="color: #BB0000;">NA</span>          -<span style="color: #BB0000;">0.000</span><span style="color: #BB0000; text-decoration: underline;">726</span> 52<span style="text-decoration: underline;">039</span>916 52<span style="text-decoration: underline;">040</span>326 Conse…</span></span></code></pre></div>
<p>(Note that in the example dataset, some genes have missing values in
the <code>padj</code> column; this corresponds to genes that have been
filtered out via <a href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#pvaluesNA" class="external-link">independent
filtering in the DESeq2 package</a>).</p>
</div>
<div class="section level3">
<h3 id="candidate-genes">Candidate genes<a class="anchor" aria-label="anchor" href="#candidate-genes"></a>
</h3>
<p>A list of candidate genes (e.g. genes previously found associated
with a trait of interest based on literature search) can be provided as
a tibble or data-frame, with one row per gene. This data-frame can also
contain variants of interest (see below). The data-frame should contain
at least the following columns:</p>
<ul>
<li><p><code>chromosome</code>: character column, giving the ID of the
chromosome on which each gene is located;</p></li>
<li><p><code>start</code> and <code>end</code>: numeric columns, giving
the starting and end position of the gene along the chromosome (in bp).
These two columns will be used to calculate the position of the gene as
the half-way point between the start and end of the gene. For genomic
variants or markers, simply set both the <code>start</code> and
<code>end</code> columns to the physical position of the
marker.</p></li>
<li><p><code>name</code>: character column, giving the name of the
candidate gene that will be displayed in the HIDECAN plot. Set to
<code>NA</code> for any subset of genes to remove their label in the
plot (can help if many genes are very close, to avoid cluttering the
plot).</p></li>
</ul>
<p>Any other column present in the data-frame will be ignored. An
example of valid input is shown below:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 6</span></span></span>
<span><span class="co">#&gt;   id                   chromosome    start      end name       gene_name        </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGSC0003DMG400003155 ST4.03ch03 46<span style="text-decoration: underline;">757</span>152 46<span style="text-decoration: underline;">762</span>127 4CL        4-coumarate-CoA …</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PGSC0003DMG400014223 ST4.03ch03 57<span style="text-decoration: underline;">466</span>692 57<span style="text-decoration: underline;">469</span>946 4CL2       4-coumarate-CoA …</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PGSC0003DMG400011189 ST4.03ch07  1<span style="text-decoration: underline;">001</span>854  1<span style="text-decoration: underline;">006</span>278 HQT        HQT              </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PGSC0003DMG400005492 ST4.03ch05 36<span style="text-decoration: underline;">342</span>746 36<span style="text-decoration: underline;">347</span>409 PAL        phenylalanine am…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PGSC0003DMG400005279 ST4.03ch05 42<span style="text-decoration: underline;">523</span>943 42<span style="text-decoration: underline;">525</span>912 peroxidase peroxidase       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PGSC0003DMG400007782 ST4.03ch03 38<span style="text-decoration: underline;">537</span>202 38<span style="text-decoration: underline;">540</span>209 PHO1A      PHO1A</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="creating-a-hidecan-plot">Creating a HIDECAN plot<a class="anchor" aria-label="anchor" href="#creating-a-hidecan-plot"></a>
</h2>
<p>The <code><a href="../reference/hidecan_plot.html">hidecan_plot()</a></code> function creates a HIDECAN plot. It
takes as input the data-frames presented above, as well as the score and
log2(fold-change) thresholds used to select the significant markers and
genes.</p>
<p>In this example, we only show QTL regions and markers with a score
above 4, which corresponds to a p-value below
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>1</mn><mo>×</mo><msup><mn>10</mn><mrow><mo>−</mo><mn>4</mn></mrow></msup></mrow><annotation encoding="application/x-tex">1\times10^{-4}</annotation></semantics></math>,
and genes with a score above 1.3, which corresponds to a p-value of
0.05. We don’t place any threshold on the log2(fold-change) of the
genes:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  qtl_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"QTL"</span><span class="op">]</span><span class="op">]</span>,           <span class="co">## data-frame of GWAS results  </span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,          <span class="co">## data-frame of GWAS results          </span></span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,              <span class="co">## data-frame of DE results              </span></span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,            <span class="co">## data-frame of candidate genes             </span></span>
<span>  score_thr_qtl <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,   <span class="co">## sign. threshold for QTL           </span></span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,  <span class="co">## sign. threshold for GWAS</span></span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,      <span class="co">## sign. threshold for DE</span></span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>                    <span class="co">## log2FC threshold for DE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-plot-1.png" width="1008"></p>
<p>Note that it is possible to provide only a subset of the possible
input data, e.g. only GWAS results and a list of candidate genes:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,          </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,            </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="fl">4</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-plot-gwas-can-only-1.png" width="1008"></p>
</div>
<div class="section level2">
<h2 id="removing-empty-chromosomes">Removing empty chromosomes<a class="anchor" aria-label="anchor" href="#removing-empty-chromosomes"></a>
</h2>
<p>By default, the HIDECAN plot shows all chromosomes present in the
input data. However, it is possible that some of the chromosomes appear
empty, as they do not contain any significant region, marker or gene,
nor any candidate gene. In this case, it is possible to exclude such
“empty” chromosomes from the HIDECAN plot, through the
<code>remove_empty_chrom</code> argument.</p>
<p>We will demonstrate that by increasing the score threshold applied to
the GWAS results, in order to get fewer significant markers. In this
case, chromosomes 0, 6, 9 and 10 do not contain any significant marker
of gene of interest:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Chromosomes 0, 6, 9 and 10 are empty</span></span>
<span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,          </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,            </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="fl">5</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-plot-with-empty-chrom-1.png" width="1008"></p>
<p>By setting the <code>remove_empty_chrom</code> argument to
<code>TRUE</code>, these chromosomes will be removed from the plot:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,          </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,            </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  remove_empty_chrom <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-plot-without-empty-chrom-1.png" width="1008"></p>
</div>
<div class="section level2">
<h2 id="selecting-chromosomes-and-genomic-positions">Selecting chromosomes and genomic positions<a class="anchor" aria-label="anchor" href="#selecting-chromosomes-and-genomic-positions"></a>
</h2>
<p>It is possible to specify which chromosomes should be represented in
the HIDECAN plot, via the <code>chroms</code> argument. For example,
with the following command we restrict the plot to chromosomes 7 and
8:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  qtl_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"QTL"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,                    </span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,                          </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>, </span>
<span>  score_thr_qtl <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,                  </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,  </span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,      </span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  chroms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ST4.03ch07"</span>, <span class="st">"ST4.03ch08"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-select-chroms-1.png" width="1008"></p>
<p>We can also “zoom in” on some or all chromosomes, through the
<code>chrom_limits</code> argument. To zoom in on all chromosomes at
once, we pass to the <code>chrom_limits</code> argument an integer
vector of length 2, which gives the lower and upper limits in bp to use.
In that case, the QTL regions will be cropped to these limits as well.
For example here, we focus on the 10-20Mb region of each chromosome:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  qtl_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"QTL"</span><span class="op">]</span><span class="op">]</span>,   </span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,                    </span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,                          </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  score_thr_qtl <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,  </span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,      </span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  chrom_limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10e6</span>, <span class="fl">20e6</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-chrom-limits-all-chrom-1.png" width="1008"></p>
<p>Alternatively, we can apply different limits to some of the
chromosomes, by passing a named list to the argument. The names of the
list should match the chromosomes name, and each element should be an
integer vector of length 2 giving the lower and upper limits in bp to
use for the corresponding chromosome. For example, we will focus on the
10-20Mb region for chromosome 1, and the 30-40Mb region for chromosome
5, and leave all other chromosomes as is:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  qtl_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"QTL"</span><span class="op">]</span><span class="op">]</span>, </span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,                    </span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,                          </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,                  </span>
<span>  score_thr_qtl <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,                  </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,  </span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,      </span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  chrom_limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"ST4.03ch01"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10e6</span>, <span class="fl">20e6</span><span class="op">)</span>,</span>
<span>                      <span class="st">"ST4.03ch05"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">30e6</span>, <span class="fl">40e6</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-chrom-limits-some-chrom-1.png" width="1008"></p>
<p>The two options <code>chroms</code> and <code>chrom_limits</code> can
be used together:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  qtl_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"QTL"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,                    </span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,                          </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,                   </span>
<span>  score_thr_qtl <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,                 </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,  </span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,      </span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  chroms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ST4.03ch07"</span>, <span class="st">"ST4.03ch08"</span><span class="op">)</span>,</span>
<span>  chrom_limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"ST4.03ch07"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">50e6</span>, <span class="fl">55e6</span><span class="op">)</span>,</span>
<span>                      <span class="st">"ST4.03ch08"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">45e6</span>, <span class="fl">50e6</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-select-chroms-and-chrom-lims-1.png" width="1008"></p>
</div>
<div class="section level2">
<h2 id="colour-genes-by-log2fold-change">Colour genes by log2(fold-change)<a class="anchor" aria-label="anchor" href="#colour-genes-by-log2fold-change"></a>
</h2>
<p>By default, in a HIDECAN plot, the points representing both
significant markers and DE genes are coloured according to their GWAS/DE
score. However, it is possible to colour the DE genes by their
log2(fold-change) value instead, by setting the
<code>colour_genes_by_score</code> argument to <code>FALSE</code>:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,          </span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,              </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,              </span>
<span>  score_thr_qtl <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,          </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,</span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  colour_genes_by_score <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-genes-colour-log2fc-1.png" width="1008"></p>
<p>Genes with a negative log2(fold-change) will be represented with a
shade of blue, and genes with a positive log2(fold-change) will be
represented with a shade of red.</p>
</div>
<div class="section level2">
<h2 id="more-than-one-qtl-gwas-de-or-candidate-gene-list">More than one QTL, GWAS, DE or candidate gene list<a class="anchor" aria-label="anchor" href="#more-than-one-qtl-gwas-de-or-candidate-gene-list"></a>
</h2>
<p>The <code><a href="../reference/hidecan_plot.html">hidecan_plot()</a></code> function can take as an input lists
of data-frames for QTL mapping, GWAS and DE results or candidate genes.
This way, it is possible to visualise more than one QTL mapping, GWAS or
DE analyses at once, for example if investigating several traits at once
or comparing more than two treatment groups.</p>
<p>For this example, we’ll focus on chromosomes 7 and 8 (only for
clarity of the plot):</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://purrr.tidyverse.org/" class="external-link">purrr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Retaining only markers and genes on chromosomes 7 and 8</span></span>
<span><span class="va">x_small</span> <span class="op">&lt;-</span> <span class="va">x</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="op">~</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">.x</span>, <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">chromosome</span>, <span class="st">"(07|08)"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>We’ll create a second data-frame of GWAS results by shuffling the
marker scores in the example dataset:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Creating a second GWAS result tibble by shuffling </span></span>
<span><span class="co">## the marker scores from the original data</span></span>
<span><span class="va">gwas_1</span> <span class="op">&lt;-</span> <span class="va">x_small</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">gwas_2</span> <span class="op">&lt;-</span> <span class="va">gwas_1</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">score</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>We can pass both GWAS results data-frames to the
<code><a href="../reference/hidecan_plot.html">hidecan_plot()</a></code> function as a list:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">gwas_1</span>, <span class="va">gwas_2</span><span class="op">)</span>,</span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,</span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-multiple-gwas-input-1.png" width="1008"></p>
<p>By default, the two GWAS tracks will be given unique y-axis labels,
as can be seen above. It is possible to customise this by naming the
elements in the input list:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  gwas_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Trait 1"</span> <span class="op">=</span> <span class="va">gwas_1</span>, </span>
<span>                   <span class="st">"Trait 2"</span> <span class="op">=</span> <span class="va">gwas_2</span><span class="op">)</span>,</span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,</span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/multiple-gwas-input-named-1.png" width="1008"></p>
<p>If you want to represent different sets of QTL regions which would be
coloured by track rather than score, see the example at the end of the
<a href="hidecan.html#controlling-the-aesthetics-colours-point-shapes-labels-of-the-tracks">“Controlling
the aesthetics”</a> section.</p>
</div>
<div class="section level2">
<h2 id="defining-chromosomes-length">Defining chromosomes length<a class="anchor" aria-label="anchor" href="#defining-chromosomes-length"></a>
</h2>
<p>By default, the <code><a href="../reference/hidecan_plot.html">hidecan_plot()</a></code> function calculates the
length of the different chromosomes based on the input data, by looking
at the maximum position of genes and markers on each chromosome.
However, it is also possible to pass on a tibble of chromosome length
(in bp) through the <code>chrom_length</code> argument.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Chromosomes length as recorded in Ensembl Plants</span></span>
<span><span class="va">potato_chrom_length</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  ST4.03ch00 <span class="op">=</span> <span class="fl">45813526</span>,</span>
<span>  ST4.03ch01 <span class="op">=</span> <span class="fl">88663952</span>,</span>
<span>  ST4.03ch02 <span class="op">=</span> <span class="fl">48614681</span>,</span>
<span>  ST4.03ch03 <span class="op">=</span> <span class="fl">62190286</span>,</span>
<span>  ST4.03ch04 <span class="op">=</span> <span class="fl">72208621</span>,</span>
<span>  ST4.03ch05 <span class="op">=</span> <span class="fl">52070158</span>,</span>
<span>  ST4.03ch06 <span class="op">=</span> <span class="fl">59532096</span>,</span>
<span>  ST4.03ch07 <span class="op">=</span> <span class="fl">56760843</span>,</span>
<span>  ST4.03ch08 <span class="op">=</span> <span class="fl">56938457</span>,</span>
<span>  ST4.03ch09 <span class="op">=</span> <span class="fl">61540751</span>,</span>
<span>  ST4.03ch10 <span class="op">=</span> <span class="fl">59756223</span>,</span>
<span>  ST4.03ch11 <span class="op">=</span> <span class="fl">45475667</span>,</span>
<span>  ST4.03ch12 <span class="op">=</span> <span class="fl">61165649</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="co">## turn a named vector into a tibble</span></span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/enframe.html" class="external-link">enframe</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"chromosome"</span>,</span>
<span>          value <span class="op">=</span> <span class="st">"length"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">potato_chrom_length</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 2</span></span></span>
<span><span class="co">#&gt;   chromosome   length</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> ST4.03ch00 45<span style="text-decoration: underline;">813</span>526</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> ST4.03ch01 88<span style="text-decoration: underline;">663</span>952</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> ST4.03ch02 48<span style="text-decoration: underline;">614</span>681</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> ST4.03ch03 62<span style="text-decoration: underline;">190</span>286</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> ST4.03ch04 72<span style="text-decoration: underline;">208</span>621</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> ST4.03ch05 52<span style="text-decoration: underline;">070</span>158</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  qtl_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"QTL"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,          </span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,              </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  score_thr_qtl <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,</span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  chrom_length <span class="op">=</span> <span class="va">potato_chrom_length</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-chrom-length-1.png" width="1008"></p>
<p>Note that in this case we can’t really see the difference with the
computed chromosome length values.</p>
</div>
<div class="section level2">
<h2 id="controlling-the-plot-properties">Controlling the plot properties<a class="anchor" aria-label="anchor" href="#controlling-the-plot-properties"></a>
</h2>
<p>The <code><a href="../reference/hidecan_plot.html">hidecan_plot()</a></code> function offers several arguments to
control different aspects of the HIDECAN plot. For example, it is
possible to specify the number of rows or columns the plot should have,
through the <code>n_rows</code> and <code>n_cols</code> arguments. Note
that only one of these arguments will be considered (<code>n_rows</code>
takes precedence):</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Specifying the number of rows</span></span>
<span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  qtl_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"QTL"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,          </span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,              </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,             </span>
<span>  score_thr_qtl <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,           </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,</span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  n_rows <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-nrows-1.png" width="1008"></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Specifying the number of columns</span></span>
<span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  qtl_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"QTL"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,          </span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,              </span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,            </span>
<span>  score_thr_qtl <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,            </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.005</span><span class="op">)</span>,</span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  n_cols <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/hidecan-ncols-1.png" width="1008"></p>
<p>In addition, it is possible to:</p>
<ul>
<li><p>add a title and subtitle to the plot (<code>title</code> and
<code>subtitle</code> arguments);</p></li>
<li><p>control the position of the legend (<code>legend_position</code>
argument);</p></li>
<li><p>control the size of the points (<code>point_size</code>
argument);</p></li>
<li><p>control the size of the labels and padding around the text for
the candidate genes labels (<code>label_size</code> and
<code>label_padding</code> arguments).</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="creating-a-manhattan-plot">Creating a Manhattan plot<a class="anchor" aria-label="anchor" href="#creating-a-manhattan-plot"></a>
</h2>
<p>It is possible to visualise GWAS results with a Manhattan plot
through the <code><a href="../reference/manhattan_plot.html">manhattan_plot()</a></code> function. The function takes
as input a data-frame (or list of data-frames) of GWAS results, which
must follow the same rules as for <code>plot_hidecan()</code> function
(see <a href="hidecan.html#gwas-results">this section</a> of the
vignette). A horizontal line representing the significance threshold can
be added to the plot, by passing the threshold value to the
<code>score_thr</code> argument.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/manhattan_plot.html">manhattan_plot</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>, score_thr <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/manhattan-plot-1.png" width="1056"></p>
<p>As for the <code><a href="../reference/hidecan_plot.html">hidecan_plot()</a></code> function, it is possible to
pass a list of GWAS results to the function, in order to draw several
Manhattan plots in one figure. The names of the list will be used to
identify the different plots:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/manhattan_plot.html">manhattan_plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Trait A"</span> <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                    <span class="st">"Trait B"</span> <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, </span>
<span>               score_thr <span class="op">=</span> <span class="fl">4</span>,</span>
<span>               ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/manhattan-plot-facets-1.png" width="1056"></p>
</div>
<div class="section level2">
<h2 id="controlling-the-aesthetics-colours-point-shapes-labels-of-the-tracks">Controlling the aesthetics (colours, point shapes, labels) of the
tracks<a class="anchor" aria-label="anchor" href="#controlling-the-aesthetics-colours-point-shapes-labels-of-the-tracks"></a>
</h2>
<p>The default look of the different data types (QTL regions, GWAS
peaks, DE genes and candidate genes) in a HIDECAN plots is controlled by
the arguments returned by <code><a href="../reference/hidecan_aes.html">hidecan_aes()</a></code>. More specifically,
the function controls:</p>
<ul>
<li><p>whether each track is represented as a point (the default for
GWAS, DE and candidate genes tracks) or as a rectangle spanning a
genomic region (the default for QTL track),</p></li>
<li><p>the label of each data type on the x-axis (e.g. ‘GWAS peaks’ for
GWAS results),</p></li>
<li><p>the colour of the vertical lines used to showcase the position of
markers or genes, or of the vertical “shadow” used to showcase the
position of genomic regions,</p></li>
<li><p>the shape of the points,</p></li>
<li><p>for genomic regions, the width of the box used to represent
them,</p></li>
<li><p>whether the name of the regions/markers/genes are displayed (by
default, will only show for candidate genes),</p></li>
<li><p>the colour palette used for the score or log2 fold-change of the
points.</p></li>
</ul>
<p>The default values used for these parameters have been chosen to
ensure that the resulting plot is easily understandable. However, it is
possible to change them. To do so, we’ll start by saving the defaults
returned by <code><a href="../reference/hidecan_aes.html">hidecan_aes()</a></code>:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">default_aes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hidecan_aes.html">hidecan_aes</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">default_aes</span>, max.level <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 5</span></span>
<span><span class="co">#&gt;  $ QTL_data_thr   :List of 7</span></span>
<span><span class="co">#&gt;  $ GWAS_data_thr  :List of 7</span></span>
<span><span class="co">#&gt;  $ DE_data_thr    :List of 7</span></span>
<span><span class="co">#&gt;  $ CAN_data_thr   :List of 7</span></span>
<span><span class="co">#&gt;  $ CUSTOM_data_thr:List of 7</span></span>
<span></span>
<span><span class="va">default_aes</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; $y_label</span></span>
<span><span class="co">#&gt; [1] "QTL regions"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $show_as_rect</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $line_colour</span></span>
<span><span class="co">#&gt; [1] "grey80"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $point_shape</span></span>
<span><span class="co">#&gt; [1] 24</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $show_name</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $fill_scale</span></span>
<span><span class="co">#&gt; &lt;ScaleContinuous&gt;</span></span>
<span><span class="co">#&gt;  Range:  </span></span>
<span><span class="co">#&gt;  Limits:    0 --    1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $rect_width</span></span>
<span><span class="co">#&gt; [1] 0.5</span></span></code></pre></div>
<p>As we can see above, the function returns a list of the default
aesthetics to use for QTL mapping results (the <code>QTL_data_thr</code>
element), for GWAS results (the <code>GWAS_data_thr</code> element), for
DE results (<code>DE_data_thr</code> element), for candidate genes
(<code>CAN_data_thr</code> element), and for custom tracks
(<code>CUSTOM_data_thr</code> element - see the <a href="web_only/custom_tracks.html">Adding custom tracks</a> article for
more information). Each element is itself a list, containing the value
for each attribute mentioned above.</p>
<p>Let’s modify the label given to the GWAS peaks, and the colour
palette used for DE score:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://sjmgarnier.github.io/viridis/" class="external-link">viridis</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">default_aes</span><span class="op">$</span><span class="va">GWAS_data_thr</span><span class="op">$</span><span class="va">y_label</span> <span class="op">&lt;-</span> <span class="st">"GWAS signif. markers"</span></span>
<span></span>
<span><span class="va">default_aes</span><span class="op">$</span><span class="va">DE_data_thr</span><span class="op">$</span><span class="va">fill_scale</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://sjmgarnier.github.io/viridis/reference/scale_viridis.html" class="external-link">scale_fill_viridis</a></span><span class="op">(</span></span>
<span>  <span class="st">"DE gene score"</span>,</span>
<span>  option <span class="op">=</span> <span class="st">"inferno"</span>,</span>
<span>  guide <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_colourbar.html" class="external-link">guide_colourbar</a></span><span class="op">(</span></span>
<span>    title.position <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>    title.hjust <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>    order <span class="op">=</span> <span class="fl">4</span> <span class="co"># ensures this legend is shown after the one for GWAS scores</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>We can use these modified values by passing this list to
<code><a href="../reference/hidecan_plot.html">hidecan_plot()</a></code> through its <code>custom_aes</code>
argument:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  qtl_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"QTL"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,        </span>
<span>  score_thr_qtl <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,  </span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,</span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  custom_aes <span class="op">=</span> <span class="va">default_aes</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/use-custom-aes-1.png" width="1056"></p>
<p>We can use this trick to represent different set of QTL regions,
e.g. associated with different traits coloured by track rather than by
their score. Let’s start by generating two other random set of QTL
regions:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">356</span><span class="op">)</span></span>
<span><span class="va">qtl_1</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[[</span><span class="st">"QTL"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">qtl_2</span> <span class="op">&lt;-</span> <span class="va">qtl_1</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice_sample</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>    start <span class="op">=</span> <span class="va">start</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="fl">3</span>, min <span class="op">=</span> <span class="op">-</span><span class="fl">3</span>, max <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">*</span> <span class="fl">1e6</span><span class="op">)</span>,</span>
<span>    end <span class="op">=</span> <span class="va">end</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="fl">3</span>, min <span class="op">=</span> <span class="op">-</span><span class="fl">3</span>, max <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">*</span> <span class="fl">1e6</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">qtl_3</span> <span class="op">&lt;-</span> <span class="va">qtl_1</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice_sample</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>    start <span class="op">=</span> <span class="va">start</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="fl">3</span>, min <span class="op">=</span> <span class="op">-</span><span class="fl">3</span>, max <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">*</span> <span class="fl">1e6</span><span class="op">)</span>,</span>
<span>    end <span class="op">=</span> <span class="va">end</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="fl">3</span>, min <span class="op">=</span> <span class="op">-</span><span class="fl">3</span>, max <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">*</span> <span class="fl">1e6</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>The key is to give each set of QTL regions a score representing the
colour we want to assign to them (here, we’ll use a different colour for
each trait):</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qtl_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="st">"Trait 1"</span> <span class="op">=</span> <span class="va">qtl_1</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>score <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>  <span class="st">"Trait 2"</span> <span class="op">=</span> <span class="va">qtl_2</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>score <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,</span>
<span>  <span class="st">"Trait 3"</span> <span class="op">=</span> <span class="va">qtl_3</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>score <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>We can then create a specific colour palette for each trait:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">new_aes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hidecan_aes.html">hidecan_aes</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">new_aes</span><span class="op">$</span><span class="va">QTL_data_thr</span><span class="op">$</span><span class="va">fill_scale</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_fill_gradientn</a></span><span class="op">(</span></span>
<span>  colours <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"aliceblue"</span>, <span class="st">"cornflowerblue"</span>, <span class="st">"mediumslateblue"</span><span class="op">)</span>,</span>
<span>  guide <span class="op">=</span> <span class="st">"none"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  qtl_list <span class="op">=</span> <span class="va">qtl_list</span>,</span>
<span>  score_thr_qtl <span class="op">=</span> <span class="fl">0</span>,                  <span class="co">## to avoid removing QTLs based on score</span></span>
<span>  chrom_length <span class="op">=</span> <span class="va">potato_chrom_length</span>,</span>
<span>  remove_empty_chrom <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  custom_aes <span class="op">=</span> <span class="va">new_aes</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hidecan_files/figure-html/show-hidecan-multiple-qtls-1.png" width="1008"></p>
</div>
<div class="section level2">
<h2 id="viewport-error">Viewport error<a class="anchor" aria-label="anchor" href="#viewport-error"></a>
</h2>
<p>If you are working on RStudio, you may encounter the following
error:</p>
<pre><code><span><span class="co">#&gt; Error in grid.Call(C_convert, x, as.integer(whatfrom), as.integer(whatto), : Viewport has zero dimension(s)</span></span></code></pre>
<p>This is caused by the plotting window being too small. Try increasing
the size of the plotting window in the RStudio console. Alternatively,
you can save the plot into an R object, then use
<code><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggplot2::ggsave()</a></code> to save it into a file:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hidecan_plot.html">hidecan_plot</a></span><span class="op">(</span></span>
<span>  qtl_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"QTL"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  gwas_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  de_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"DE"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  can_list <span class="op">=</span> <span class="va">x</span><span class="op">[[</span><span class="st">"CAN"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  score_thr_qtl <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_gwas <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.0001</span><span class="op">)</span>,</span>
<span>  score_thr_de <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,</span>
<span>  log2fc_thr <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave</a></span><span class="op">(</span><span class="st">"hidecan_plot.pdf"</span>, <span class="va">p</span>, width <span class="op">=</span> <span class="fl">10.5</span>, height <span class="op">=</span> <span class="fl">11</span><span class="op">)</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Olivia Angelin-Bonnet.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
